// @jappyjan/even-better-sdk/dist/index.js
const i = L;
(function(o, e) {
  const t = L, n = o();
  for (; ; )
    try {
      if (parseInt(t(375)) / 1 + parseInt(t(675)) / 2 * (-parseInt(t(908)) / 3) + parseInt(t(1092)) / 4 + parseInt(t(391)) / 5 + -parseInt(t(681)) / 6 + -parseInt(t(824)) / 7 * (parseInt(t(1084)) / 8) + parseInt(t(889)) / 9 * (-parseInt(t(554)) / 10) === e) break;
      n.push(n.shift());
    } catch {
      n.push(n.shift());
    }
})(rt, 136384);
var At = Object[i(434)], qt = Object[i(1149)], pt = Object[i(756)], at = Object[i(1032)], Mt = Object[i(1171)][i(500)], Pt = Object[i(1171)][i(1e3)], lt = (o, e, t) => e in o ? At(o, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : o[e] = t, D = (o, e) => {
  const t = i, n = { QACgG: function(r, u, s, c) {
    return r(u, s, c);
  }, mClkz: function(r, u) {
    return r(u);
  } };
  for (var x in e || (e = {})) Mt[t(485)](e, x) && n[t(546)](lt, o, x, e[x]);
  if (at)
    for (var x of n[t(584)](at, e)) Pt[t(485)](e, x) && n[t(546)](lt, o, x, e[x]);
  return o;
}, A = (o, e) => qt(o, pt(e)), m = (o, e, t) => new Promise((n, x) => {
  const r = i, u = { yNyIG: function(d, a) {
    return d !== a;
  }, GVShu: r(926), ZJXxu: r(1005), siKxG: function(d, a) {
    return d(a);
  }, WEKez: function(d, a) {
    return d == a;
  }, QKpny: function(d, a) {
    return d == a;
  }, dzXzj: function(d, a, f) {
    return d(a, f);
  }, hrUmw: function(d, a, f) {
    return d(a, f);
  }, vOuFU: function(d, a) {
    return d != a;
  }, YDxfG: function(d, a) {
    return d == a;
  }, ZNttT: function(d, a) {
    return d(a);
  }, xzFAs: function(d, a) {
    return d === a;
  }, iZpzU: r(1081), ebvwJ: r(1101), ufAIu: function(d, a) {
    return d(a);
  }, IBIKY: function(d, a) {
    return d(a);
  } };
  var s = (d) => {
    const a = r;
    try {
      if (u[a(452)](u.GVShu, u[a(644)])) u.siKxG(l, t.next(d));
      else return _0x1d9490[a(830)]((void 0)[a(816)]);
    } catch (f) {
      u[a(921)](x, f);
    }
  }, c = (d) => {
    const a = r;
    try {
      u[a(429)](l, t[a(543)](d));
    } catch (z) {
      if (u[a(572)](u[a(1011)], u[a(1147)])) {
        var f, w, b;
        let K = (f = u[a(455)](_0x57dfbd, null) ? void 0 : _0x24e58d.sn) != null ? f : "";
        return new _0x218126({ model: _0x266766[a(636)]((w = u[a(725)](_0x2730d3, null) ? void 0 : _0x3bdb0a[a(816)]) != null ? w : ""), sn: K, status: _0x423234.fromJson(u[a(1020)](_0x18967b, u.hrUmw(_0x332af0, {}, u.vOuFU(b = u[a(347)](_0x22b409, null) ? void 0 : _0x432930.status, null) ? b : {}), { sn: K })) });
      } else u[a(963)](x, z);
    }
  }, l = (d) => d[r(1119)] ? n(d[r(862)]) : Promise[r(1126)](d[r(862)]).then(s, c);
  u.IBIKY(l, (t = t[r(934)](o, e)).next());
}), Kt = ((o) => (o[i(1026)] = i(1080), o[i(757)] = "getGlassesInfo", o[i(1190)] = i(1196), o[i(600)] = i(1034), o[i(709)] = "createStartUpPageContainer", o.RebuildPageContainer = i(390), o[i(1194)] = i(646), o.TextContainerUpgrade = i(1198), o[i(1069)] = "audioControl", o[i(541)] = i(501), o[i(430)] = "shutDownPageContainer", o))(Kt || {}), Et = ((o) => (o[i(581)] = i(1199), o[i(616)] = i(925), o))(Et || {}), jt = ((o) => (o.BridgeReady = "evenAppBridgeReady", o[i(567)] = i(698), o[i(586)] = i(759), o[i(588)] = i(848), o))(jt || {}), Bt = class ut {
  constructor(e) {
    const t = i;
    this[t(656)] = e[t(656)], this[t(565)] = e[t(565)], this[t(1025)] = e[t(1025)], this[t(669)] = e[t(669)];
  }
  toJson() {
    const e = i;
    return { uid: this.uid, name: this[e(565)], avatar: this[e(1025)], country: this[e(669)] };
  }
  static [i(916)](e) {
    const t = i, n = { PfWMe: function(c, l) {
      return c != l;
    }, MlZyg: function(c, l) {
      return c == l;
    }, LiifW: function(c, l) {
      return c == l;
    } };
    var x, r, u, s;
    return new ut({ uid: n[t(651)](x = e?.[t(656)], null) ? x : 0, name: n[t(651)](r = n.MlZyg(e, null) ? void 0 : e[t(565)], null) ? r : "", avatar: (u = n[t(1118)](e, null) ? void 0 : e[t(1025)]) != null ? u : "", country: (s = n[t(1118)](e, null) ? void 0 : e[t(669)]) != null ? s : "" });
  }
  static [i(864)]() {
    return new ut({ uid: 0, name: "", avatar: "", country: "" });
  }
}, $ = ((o) => (o[i(405)] = i(1031), o.Connecting = "connecting", o[i(803)] = i(941), o[i(1188)] = "disconnected", o[i(809)] = "connectionFailed", o))($ || {});
((o) => {
  const e = i, t = { SyamD: function(x, r) {
    return x(r);
  }, KbolU: function(x, r) {
    return x !== r;
  }, MzLNO: e(666), dzDYd: e(788), ciUBV: e(1031), RCifs: e(768), OupbA: e(941), RfCgP: e(561), eHrLX: e(388) };
  function n(x) {
    const r = e;
    if (t[r(425)](t[r(806)], t[r(362)])) switch (x) {
      case t[r(841)]:
        return t[r(841)];
      case t.RCifs:
        return t[r(697)];
      case r(941):
        return t[r(678)];
      case t[r(894)]:
        return t[r(894)];
      case t.eHrLX:
        return t[r(751)];
      default:
        return r(1031);
    }
    else try {
      t.SyamD(_0x25d57e, _0x2bced1.next(_0x2c8b27));
    } catch (u) {
      t[r(614)](_0x4db49f, u);
    }
  }
  o[e(636)] = n;
})($ || ($ = {}));
var it = class st {
  constructor(e) {
    const t = i, n = { WnlzW: function(l, d) {
      return l != d;
    }, XcWGU: t(1031), nrmZB: function(l, d) {
      return l != d;
    } };
    var x, r, u, s, c;
    this.sn = e.sn, this[t(974)] = n[t(693)](x = e.connectType, null) ? x : n[t(1068)], this[t(708)] = (r = e[t(708)]) != null ? r : !1, this[t(595)] = n[t(499)](u = e[t(595)], null) ? u : 0, this.isCharging = n.nrmZB(s = e.isCharging, null) ? s : !1, this.isInCase = n[t(499)](c = e[t(1074)], null) ? c : !1;
  }
  [i(1128)]() {
    const e = i;
    return { sn: this.sn, connectType: this[e(974)], isWearing: this[e(708)], batteryLevel: this[e(595)], isCharging: this[e(407)], isInCase: this[e(1074)] };
  }
  [i(438)]() {
    const e = i, t = { JHIWr: "none" };
    return this[e(974)] === t[e(423)];
  }
  isConnected() {
    const e = i, t = { aBhTi: function(n, x) {
      return n === x;
    }, GJsAg: e(941) };
    return t[e(1085)](this[e(974)], t[e(754)]);
  }
  [i(354)]() {
    const e = i, t = { dNCEG: e(768) };
    return this[e(974)] === t[e(1103)];
  }
  isDisconnected() {
    const e = i, t = { MPmwV: function(n, x) {
      return n === x;
    }, HuiyI: "disconnected" };
    return t[e(818)](this[e(974)], t[e(525)]);
  }
  [i(951)]() {
    const e = i, t = { MkBBo: function(n, x) {
      return n === x;
    }, eWYay: e(388) };
    return t[e(615)](this.connectType, t.eWYay);
  }
  static [i(916)](e) {
    const t = i, n = { aHRcX: function(d, a) {
      return d != a;
    }, VhyuY: function(d, a) {
      return d == a;
    }, rfXUB: function(d, a) {
      return d == a;
    }, tLWQy: function(d, a) {
      return d != a;
    }, muLsG: function(d, a) {
      return d != a;
    } };
    var x, r, u, s, c, l;
    return new st({ sn: n[t(880)](x = e?.sn, null) ? x : "", connectType: $[t(636)]((r = n[t(838)](e, null) ? void 0 : e[t(974)]) != null ? r : ""), isWearing: n.aHRcX(u = n[t(353)](e, null) ? void 0 : e[t(708)], null) ? u : !1, batteryLevel: (s = n[t(353)](e, null) ? void 0 : e[t(595)]) != null ? s : 0, isCharging: n[t(1057)](c = n[t(838)](e, null) ? void 0 : e[t(407)], null) ? c : !1, isInCase: n[t(533)](l = n.rfXUB(e, null) ? void 0 : e.isInCase, null) ? l : !1 });
  }
  static [i(864)](e = "") {
    const t = i, n = { JtZGw: t(1031) };
    return new st({ sn: e, connectType: n[t(976)], isWearing: !1, batteryLevel: 0, isCharging: !1, isInCase: !1 });
  }
}, E = ((o) => (o.G1 = "g1", o.G2 = "g2", o[i(1106)] = i(384), o))(E || {});
((o) => {
  const e = i, t = { dodmD: function(u, s, c, l, d) {
    return u(s, c, l, d);
  }, ExJza: e(378), FBNuO: e(1174), QYbOo: e(850), TXmwP: function(u, s, c, l, d) {
    return u(s, c, l, d);
  }, WDajD: e(553), QHGyq: e(1145), NWwbf: function(u, s) {
    return u(s);
  }, fGWpB: function(u, s, c, l) {
    return u(s, c, l);
  }, YZPeU: function(u, s, c, l) {
    return u(s, c, l);
  }, uXUyy: e(1030), AApQR: "Height", UaBAZ: e(978), rJfzA: "BorderWidth", tJglb: e(363), KMint: e(1105), eXxuU: "BorderColor", xGhmS: e(812), oOmQP: function(u, s) {
    return u(s);
  }, NEZcT: e(895), DrFqJ: "Border_Radius", fxaxy: e(1153), sBMXy: e(1164), TZGUg: function(u, s) {
    return u(s);
  }, USxrl: function(u, s, c, l, d) {
    return u(s, c, l, d);
  }, VpnSO: e(870), jAoYo: e(676), QKZRd: function(u, s, c, l, d) {
    return u(s, c, l, d);
  }, LtiLA: e(372), tLecq: e(1086), TOaEv: e(377), DiwCR: e(817), RFiJS: e(498), hmIbb: e(723), inAdK: function(u, s) {
    return u !== s;
  }, JfxmX: e(783), PKcKM: e(384), tFtdr: e(527), ZMSTy: function(u, s) {
    return u === s;
  }, AxTbA: function(u, s) {
    return u === s;
  } };
  function n(u) {
    const s = e;
    if (t.inAdK(s(394), t.JfxmX)) switch (u) {
      case "g1":
        return "g1";
      case "g2":
        return "g2";
      case t[s(451)]:
        return t[s(451)];
      default:
        return "g1";
    }
    else return new _0x33d1c4({ xPosition: _0x1fd37a(t.dodmD(_0x3e9074, _0x1099e8, t[s(422)], t[s(747)], t.QYbOo)), yPosition: _0x250353(t[s(795)](_0x4ceed3, _0x3c00d5, s(454), t[s(557)], t[s(1138)])), width: t[s(1110)](_0x429ee1, t[s(456)](_0x1fb44e, _0x24ef7e, s(526), "Width")), height: _0x5102c6(t[s(530)](_0x107209, _0xe83241, t[s(620)], t[s(446)])), borderWidth: t[s(1110)](_0x3bddb4, t[s(417)](_0x1cc30e, _0xa0e066, t[s(733)], t.rJfzA, t.tJglb)), borderColor: _0x41e9ad(t[s(417)](_0x38615e, _0x147327, t[s(1184)], t.eXxuU, t[s(459)])), borderRadius: t[s(745)](_0xc71a94, t[s(795)](_0x323b6f, _0x59cb55, s(665), t[s(617)], t[s(903)])), paddingLength: t[s(1110)](_0x313be2, _0x610f24(_0x3489b4, t[s(713)], s(836), t[s(359)])), containerID: t[s(948)](_0x99024e, t[s(519)](_0x40acda, _0x19de30, s(1156), t[s(1127)], t[s(419)])), containerName: t[s(745)](_0x2c1533, t[s(1028)](_0x5ebbde, _0x2df1c6, t[s(371)], t[s(453)], s(1094))), isEventCapture: t[s(745)](_0x54573d, _0x4e0702(_0x513384, t.TOaEv, t.DiwCR, t.RFiJS)), content: _0x103109(_0x561054(_0x53c1ad, s(448), t[s(789)])) });
  }
  o[e(636)] = n;
  function x(u) {
    const s = e;
    if (t[s(497)] === s(527)) return t[s(1049)](u, "g1") || t.AxTbA(u, "g2");
    {
      if (_0x2087a3 == null) return;
      let c = _0xbecabb(_0x14f089);
      return _0x5d112b[s(548)](c) ? c : void 0;
    }
  }
  o[e(830)] = x;
  function r(u) {
    const s = e;
    return t[s(1049)](u, s(384));
  }
  o.isRing = r;
})(E || (E = {}));
var Ht = class Dt {
  constructor(e) {
    const t = i;
    var n;
    this.model = e[t(816)], this.sn = e.sn, this.status = (n = e[t(825)]) != null ? n : it[t(864)](e.sn);
  }
  static [i(916)](e) {
    const t = i, n = { vajWN: function(c, l) {
      return c != l;
    }, iYBlF: function(c, l) {
      return c == l;
    }, bQBty: function(c, l, d) {
      return c(l, d);
    }, dnZep: function(c, l, d) {
      return c(l, d);
    }, CREAp: function(c, l) {
      return c == l;
    } };
    var x, r, u;
    let s = n[t(1202)](x = n.iYBlF(e, null) ? void 0 : e.sn, null) ? x : "";
    return new Dt({ model: E[t(636)](n[t(1202)](r = n[t(642)](e, null) ? void 0 : e[t(816)], null) ? r : ""), sn: s, status: it[t(916)](n[t(1139)](A, n[t(585)](D, {}, n[t(1202)](u = n.CREAp(e, null) ? void 0 : e[t(825)], null) ? u : {}), { sn: s })) });
  }
  [i(409)](e) {
    ({ TMbiM: function(x, r) {
      return x === r;
    } })[i(566)](e.sn, this.sn) && (this.status = e);
  }
  [i(830)]() {
    const e = i;
    return E.isGlasses(this[e(816)]);
  }
  [i(1082)]() {
    return E[i(1082)](this.model);
  }
  [i(1128)]() {
    const e = i;
    return { model: this.model, sn: this.sn, status: this[e(825)][e(1128)]() };
  }
}, Nt = ((o) => (o[i(517)] = "APP_REQUEST_CREATE_PAGE_SUCCESS", o.APP_REQUEST_CREATE_INVAILD_CONTAINER = i(1201), o[i(396)] = i(396), o[i(1053)] = i(1053), o[i(609)] = "APP_REQUEST_UPGRADE_IMAGE_RAW_DATA_SUCCESS", o.APP_REQUEST_UPGRADE_IMAGE_RAW_DATA_FAILED = i(1013), o[i(1001)] = "APP_REQUEST_REBUILD_PAGE_SUCCESS", o[i(753)] = "APP_REQUEST_REBUILD_PAGE_FAILD", o[i(508)] = "APP_REQUEST_UPGRADE_TEXT_DATA_SUCCESS", o[i(564)] = i(564), o[i(649)] = "APP_REQUEST_UPGRADE_SHUTDOWN_SUCCESS", o[i(427)] = i(427), o[i(1193)] = i(1193), o[i(469)] = "APP_REQUEST_AUDIO_CTR_SUCCESS", o[i(1178)] = "APP_REQUEST_AUDIO_CTR_FAILED", o))(Nt || {}), Ut = ((o) => (o[o.P100 = 100] = i(481), o[o[i(606)] = 200] = i(606), o[o[i(716)] = 300] = i(716), o[o[i(954)] = 400] = i(954), o[o.P500 = 500] = "P500", o[o.P600 = 600] = i(935), o[o[i(577)] = 700] = i(577), o[o[i(804)] = 800] = i(804), o[o[i(761)] = 900] = "P900", o[o[i(1163)] = 1e3] = i(1163), o))(Ut || {});
function q(o) {
  const e = i;
  return !!o && { sduKk: function(n, x) {
    return n == x;
  } }[e(1055)](typeof o, e(487)) && !Array[e(626)](o);
}
function v(o, ...e) {
  const t = i, n = { SafwW: function(x, r) {
    return x !== r;
  } };
  for (let x of e) if (o && n[t(477)](o[x], void 0)) return o[x];
}
function dt(o) {
  return { khngV: function(n, x) {
    return n(x);
  } }[i(1133)](String, o).replace(/_/g, "").toLowerCase();
}
function P(o, ...e) {
  const t = i, n = { emQYT: function(u, s, ...c) {
    return u(s, ...c);
  }, xKnLi: function(u, s) {
    return u !== s;
  }, VhEQa: function(u, s) {
    return u(s);
  }, CsoKg: function(u, s) {
    return u === s;
  }, GcAOb: function(u, s) {
    return u < s;
  } };
  let x = n[t(550)](v, o, ...e);
  if (n[t(1027)](x, void 0)) return x;
  if (!n[t(648)](q, o) || n[t(1182)](e[t(558)], 0)) return;
  let r = e[t(949)](dt);
  for (let u in o) {
    if (!Object[t(1171)][t(500)][t(485)](o, u)) continue;
    let s = dt(u);
    for (let c = 0; n[t(750)](c, r.length); c++) if (n[t(1182)](s, r[c])) return o[u];
  }
}
function h(o) {
  const e = i, t = { yVVze: function(x, r) {
    return x(r);
  } };
  if (o == null) return;
  let n = t[e(641)](Number, o);
  return Number.isFinite(n) ? n : void 0;
}
function p(o) {
  const e = i, t = { JYxhw: function(n, x) {
    return n(x);
  } };
  if (o != null) return t[e(643)](String, o);
}
function C(o, ...e) {
  const t = i, n = { Smmov: function(x, r) {
    return x(r);
  }, zxqDC: function(x, r, ...u) {
    return x(r, ...u);
  } };
  return n[t(905)](h, n[t(712)](P, o, ...e));
}
function ot(o, ...e) {
  const t = i, n = { SlcMw: function(r, u, ...s) {
    return r(u, ...s);
  }, zDglN: function(r, u) {
    return r != u;
  }, lurdd: function(r, u) {
    return r(u);
  } };
  let x = n.SlcMw(P, o, ...e);
  if (n.zDglN(x, null)) return n[t(442)](p, x);
}
function gt(o) {
  return { MdSNb: function(t, n) {
    return t(n);
  } }.MdSNb(q, o) ? o : {};
}
function Lt(o) {
  const e = i, t = { DrVgc: function(n, x) {
    return n == x;
  }, giJKi: "string" };
  if (o != null) {
    if (t[e(927)](typeof o, t[e(968)])) return o;
    if (Array[e(626)](o)) return o[e(949)]((n) => Number(n) & 255);
    if (o instanceof Uint8Array) return Array[e(439)](o);
    if (o instanceof ArrayBuffer) return Array[e(439)](new Uint8Array(o));
  }
}
var ct = class _ {
  constructor(e = {}) {
    Object.assign(this, e);
  }
  static [i(916)](e) {
    const t = i, n = { ofAHc: function(x, r) {
      return x(r);
    }, NuXMV: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, gTTna: "itemCount", SQwSQ: t(520), ZjDNR: function(x, r) {
      return x(r);
    }, NsOKX: t(445), FwZvC: function(x, r) {
      return x(r);
    }, RuLsO: t(1129), okZUj: t(950), BfjcN: t(540), keJJY: t(420), loGDJ: t(555), Vkzkd: "Item_Name" };
    return new _({ itemCount: n[t(1054)](h, n[t(506)](v, e, n[t(828)], "ItemCount", n[t(881)])), itemWidth: n[t(699)](h, v(e, "itemWidth", t(764), n[t(437)])), isItemSelectBorderEn: n[t(943)](h, n.NuXMV(v, e, n[t(596)], n[t(863)], n[t(667)])), itemName: n.NuXMV(v, e, n[t(736)], n[t(618)], n.Vkzkd) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { QcaUD: function(x, r) {
      return x instanceof r;
    } }[t(960)](e, _) ? e[t(1128)]() : new _(e).toJson() : {};
  }
  [i(1128)]() {
    return D({}, { tkyne: function(n, x) {
      return n != x;
    } }[i(538)](this, null) ? this : {});
  }
}, O = class Y {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { rrauD: t(906), FXpcs: "ItemContainer", XefCS: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, zryJW: t(378), WxNay: "X_Position", xDBMb: function(r, u) {
      return r(u);
    }, KaGHz: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, aPMkz: t(454), rAfaU: "Y_Position", pyMVw: function(r, u, s, c) {
      return r(u, s, c);
    }, CYKBS: t(526), byXcR: t(444), jYLYX: t(1030), UcEXb: t(1162), GIEKp: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, fKKzh: "borderWidth", ZKtef: t(539), ejHEp: t(363), NDMIm: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, wWguU: "borderColor", IDpxT: t(435), VTJjk: t(812), yXleB: function(r, u) {
      return r(u);
    }, dFkdq: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, UNSuq: t(895), qWsWA: t(843), iQvnj: function(r, u) {
      return r(u);
    }, YqTZs: t(1153), vNLCH: t(836), MLrnT: t(1156), tCPqD: "ContainerID", AVthJ: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, tVjau: "containerName", PQJnY: "ContainerName", bbVnf: "Container_Name", EqfwZ: function(r, u) {
      return r(u);
    }, gkJmA: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, ZdrDl: t(377), tUCQX: t(817), RvFAF: t(498) };
    let x = v(e, n[t(1170)], n.FXpcs, t(984));
    return new Y({ xPosition: h(n[t(1036)](v, e, n.zryJW, "XPosition", n[t(800)])), yPosition: n[t(1117)](h, n.KaGHz(v, e, n[t(608)], t(553), n[t(426)])), width: n.xDBMb(h, n[t(749)](v, e, n[t(381)], n.byXcR)), height: n[t(1117)](h, n[t(749)](v, e, n[t(684)], n[t(1016)])), borderWidth: n[t(1117)](h, n[t(1039)](v, e, n.fKKzh, n[t(846)], n[t(826)])), borderColor: n.xDBMb(h, n.NDMIm(v, e, n[t(885)], n.IDpxT, n[t(730)])), borderRadius: n[t(805)](h, n[t(552)](v, e, t(665), n.UNSuq, n.qWsWA)), paddingLength: n.iQvnj(h, n[t(552)](v, e, n[t(702)], n.vNLCH, t(1164))), containerID: n[t(1117)](h, n[t(603)](v, e, n[t(969)], n[t(790)], "Container_ID")), containerName: p(n[t(940)](v, e, n[t(382)], n.PQJnY, n[t(432)])), itemContainer: x ? ct[t(916)](x) : void 0, isEventCapture: n[t(1204)](h, n.gkJmA(v, e, n[t(710)], n[t(460)], n[t(997)])) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { FYjQN: function(x, r) {
      return x instanceof r;
    } }.FYjQN(e, Y) ? e[t(1128)]() : new Y(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { BQCJz: function(n, x, r) {
      return n(x, r);
    }, SQFCE: function(n, x) {
      return n != x;
    }, BECEf: function(n, x) {
      return n != x;
    } };
    return t[e(475)](A, D({}, t[e(1143)](this, null) ? this : {}), { itemContainer: t[e(799)](this, null) && this[e(906)] ? ct[e(1128)](this[e(906)]) : void 0 });
  }
}, X = class R {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static fromJson(e) {
    const t = i, n = { LwxEP: function(x, r) {
      return x(r);
    }, HKIaK: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, xPyfS: t(378), EclRa: "XPosition", lkPHn: t(850), ZbyCn: t(454), MtgFq: t(553), GSyhZ: "Y_Position", nlvqo: function(x, r) {
      return x(r);
    }, hjLEW: function(x, r, u, s) {
      return x(r, u, s);
    }, PiyAm: t(526), sKNEE: t(444), DGFTk: function(x, r) {
      return x(r);
    }, obWpO: "Height", rRkYU: function(x, r) {
      return x(r);
    }, ZlYzl: t(978), sgMlc: "BorderWidth", gdHXu: "Border_Width", IJJhK: "borderColor", DQYJa: t(435), ddlKq: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, ojNDe: t(895), aKnVO: t(843), YEmrI: t(1153), ukZuK: "PaddingLength", Apher: function(x, r) {
      return x(r);
    }, IQLlp: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, JhALZ: t(1156), cFwkh: t(870), tDjlo: function(x, r) {
      return x(r);
    }, MsNTO: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, xhCIj: t(1086), uJCWH: function(x, r) {
      return x(r);
    }, BAaID: t(817), SuMuU: t(448), FossC: "Content" };
    return new R({ xPosition: n.LwxEP(h, n[t(913)](v, e, n[t(1136)], n[t(1095)], n.lkPHn)), yPosition: n[t(743)](h, n[t(913)](v, e, n[t(923)], n[t(482)], n.GSyhZ)), width: n.nlvqo(h, n[t(544)](v, e, n[t(744)], n.sKNEE)), height: n[t(714)](h, v(e, t(1030), n.obWpO)), borderWidth: n.rRkYU(h, n[t(913)](v, e, n.ZlYzl, n[t(1130)], n[t(994)])), borderColor: h(n.HKIaK(v, e, n[t(1044)], n[t(724)], t(812))), borderRadius: h(n[t(523)](v, e, t(665), n[t(663)], n[t(1058)])), paddingLength: n[t(769)](h, v(e, n[t(876)], n[t(679)], t(1164))), containerID: n[t(734)](h, n.IQLlp(v, e, n[t(966)], n[t(421)], t(676))), containerName: n[t(612)](p, n[t(1089)](v, e, "containerName", n.xhCIj, t(1094))), isEventCapture: n[t(556)](h, n[t(913)](v, e, "isEventCapture", n[t(746)], t(498))), content: p(n[t(544)](v, e, n[t(762)], n[t(689)])) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { jrngo: function(x, r) {
      return x instanceof r;
    } }[t(860)](e, R) ? e[t(1128)]() : new R(e)[t(1128)]() : {};
  }
  toJson() {
    return D({}, { yLYCH: function(n, x) {
      return n != x;
    } }[i(670)](this, null) ? this : {});
  }
}, k = class J {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { iYPYd: t(378), vQTKm: t(850), lqqXT: function(x, r) {
      return x(r);
    }, glmwP: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, tURNK: t(454), nWmje: t(553), rrEbL: "Y_Position", dtqnG: function(x, r, u, s) {
      return x(r, u, s);
    }, CgGUO: t(526), LecFf: "Width", HEPPj: t(1030), iKuph: t(1162), nEFAn: function(x, r) {
      return x(r);
    }, GUAYZ: t(1156), UISVB: t(870), HJcEI: t(676), pAgky: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, wbiXD: t(372), LjpYC: t(1086), LXoNc: t(1094) };
    return new J({ xPosition: h(v(e, n[t(819)], "XPosition", n[t(1160)])), yPosition: n[t(1168)](h, n.glmwP(v, e, n[t(1024)], n[t(356)], n.rrEbL)), width: n[t(1168)](h, n[t(964)](v, e, n.CgGUO, n.LecFf)), height: h(n.dtqnG(v, e, n[t(473)], n.iKuph)), containerID: n[t(1033)](h, v(e, n[t(1097)], n.UISVB, n[t(852)])), containerName: n[t(1033)](p, n[t(1104)](v, e, n[t(687)], n[t(1120)], n[t(1012)])) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { zBBFj: function(x, r) {
      return x instanceof r;
    } }[t(1124)](e, J) ? e[t(1128)]() : new J(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    return D({}, { lIzkE: function(n, x) {
      return n != x;
    } }[i(801)](this, null) ? this : {});
  }
};
(class G {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { jDets: function(x, r) {
      return x(r);
    }, jAqOj: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, WCnxF: t(1156), cStuz: t(870), SPBYF: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, tfaMx: t(372), QXlmi: "Container_Name", HNkkF: t(503), YIjNl: t(1151), gpYkH: t(1017), mdvNx: function(x, r) {
      return x(r);
    }, icitt: t(1019), RhiBw: t(370), cwMqr: t(971), rYilm: "CompressMode", LrKlc: t(774), uyoIH: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, RvTZe: t(621), VZuIU: t(947), PiqPx: function(x, r) {
      return x(r);
    }, amQaQ: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, Dpugz: t(386), mgjRU: t(928), AFrBD: "Map_Fragment_Packet_Size", wDxFp: t(893), MMcLZ: t(867), muyXQ: t(904) };
    return new G({ containerID: n[t(598)](h, n.jAqOj(v, e, n[t(368)], n[t(516)], t(676))), containerName: p(n[t(914)](v, e, n[t(854)], t(1086), n[t(658)])), mapSessionId: n.jDets(h, v(e, n.HNkkF, n[t(835)], n.gpYkH)), mapTotalSize: n[t(647)](h, v(e, t(705), n[t(1098)], n[t(624)])), compressMode: h(n[t(914)](v, e, n[t(874)], n[t(415)], n.LrKlc)), mapFragmentIndex: h(n[t(401)](v, e, n[t(365)], n[t(741)], t(932))), mapFragmentPacketSize: n[t(992)](h, n[t(536)](v, e, n.Dpugz, n[t(685)], n[t(488)])), mapRawData: n[t(914)](v, e, n.wDxFp, n[t(486)], n[t(897)]) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { dmJRe: function(x, r) {
      return x instanceof r;
    } }.dmJRe(e, G) ? e.toJson() : new G(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { zKVSk: function(n, x, r) {
      return n(x, r);
    }, SXJGU: function(n, x, r) {
      return n(x, r);
    }, PLaxR: function(n, x) {
      return n != x;
    }, Obchj: function(n, x) {
      return n(x);
    }, SrvPR: function(n, x) {
      return n == x;
    } };
    return t[e(719)](A, t[e(605)](D, {}, t[e(630)](this, null) ? this : {}), { mapRawData: t[e(915)](Lt, t[e(701)](this, null) ? void 0 : this[e(893)]) });
  }
});
var Tt = class B {
  constructor(e = {}) {
    Object.assign(this, e);
  }
  static [i(916)](e) {
    const t = i, n = { AAOyo: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, xEJmd: t(722), YuqAR: t(574), OPDGz: function(r, u) {
      return r(u);
    }, zUFbI: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, TViCZ: "containerID", VUtia: t(676), DfOas: function(r, u) {
      return r(u);
    }, DKesV: t(372), gjfWl: t(1086) };
    let x = n[t(1177)](v, e, n[t(1075)], "ImageData", n[t(820)]);
    return new B({ containerID: n[t(673)](h, n[t(1150)](v, e, n[t(623)], "ContainerID", n[t(1180)])), containerName: n.DfOas(p, v(e, n[t(449)], n[t(479)], t(1094))), imageData: B[t(659)](x) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { bFUzE: function(x, r) {
      return x instanceof r;
    } }[t(765)](e, B) ? e[t(1128)]() : new B(e)[t(1128)]() : {};
  }
  static [i(659)](e) {
    const t = i, n = { qxDLa: function(x, r) {
      return x != r;
    }, oMAmA: function(x, r) {
      return x == r;
    }, AAPgn: "string" };
    if (n[t(961)](e, null)) {
      if (n[t(956)](typeof e, n[t(918)])) return e;
      if (Array[t(626)](e)) return e[t(949)]((x) => Number(x) & 255);
      if (e instanceof Uint8Array) return Array[t(439)](e);
      if (e instanceof ArrayBuffer) return Array[t(439)](new Uint8Array(e));
    }
  }
  [i(1128)]() {
    const e = i, t = { XBmlv: function(n, x, r) {
      return n(x, r);
    }, oGRgE: function(n, x) {
      return n != x;
    } };
    return t[e(638)](A, D({}, t.oGRgE(this, null) ? this : {}), { imageData: Lt(this == null ? void 0 : this[e(722)]) });
  }
}, tt = ((o) => (o[o[i(1067)] = 0] = i(1067), o[o.invalid = 1] = i(851), o[o[i(1062)] = 2] = i(1062), o[o.outOfMemory = 3] = i(395), o))(tt || {});
((o) => {
  const e = i, t = { TmiMu: function(r, u) {
    return r(u);
  }, bgcAm: e(759), ZogjY: e(1100), EKHrE: function(r, u) {
    return r == u;
  }, VxQxN: e(929), uHXUw: function(r, u) {
    return r(u);
  }, sVHUk: function(r, u) {
    return r == u;
  }, cLiSQ: e(1205), Jngcl: function(r, u) {
    return r(u);
  }, ZyGke: e(1067), fFsGE: e(851), ACtxT: e(1062), hTwaO: e(942), tRWII: "out_of_memory" };
  function n(r) {
    const u = e, s = { KSioA: function(c, l) {
      return t[L(872)](c, l);
    }, yZqTA: t[u(1142)], hXlxn: u(1004) };
    if (t[u(780)] === t.ZogjY) switch (r) {
      case 0:
        return 0;
      case 1:
        return 1;
      case 2:
        return 2;
      case 3:
        return 3;
      default:
        return 1;
    }
    else {
      if (typeof _0x5cbf2a == s[u(988)]) return () => {
      };
      let c = (l) => {
        const d = u;
        s[d(1113)](_0x178597, l[d(1203)]);
      };
      return _0x4703f.addEventListener(s.yZqTA, c), () => {
        const l = u;
        _0x3fee5c[l(953)](s[l(711)], c);
      };
    }
  }
  o[e(355)] = n;
  function x(r) {
    const u = e;
    if (t[u(1161)](typeof r, t[u(610)]) && Number[u(548)](r)) return t.uHXUw(n, r);
    if (t.sVHUk(typeof r, t[u(1189)])) {
      let s = t[u(872)](Number, r);
      if (Number[u(548)](s)) return t[u(965)](n, s);
      let c = r.trim().toLowerCase();
      switch (c[u(511)](".") ? c[u(672)](".")[u(1010)]() : c) {
        case t.ZyGke:
          return 0;
        case t[u(802)]:
          return 1;
        case t[u(496)]:
          return 2;
        case t[u(855)]:
        case t.tRWII:
          return 3;
        default:
          return 1;
      }
    }
    if (r && t[u(886)](typeof r, u(487))) {
      if (t[u(1161)](typeof r[u(792)], u(929))) return t[u(872)](n, r[u(792)]);
      if (t.EKHrE(typeof r[u(862)], t[u(610)])) return n(r[u(862)]);
      if (t.EKHrE(typeof r[u(565)], t[u(1189)])) return t[u(965)](x, r[u(565)]);
    }
    return 1;
  }
  o.normalize = x;
})(tt || (tt = {}));
var et = ((o) => (o[i(1067)] = "success", o[i(1169)] = "imageException", o[i(450)] = "imageSizeInvalid", o.imageToGray4Failed = i(936), o.sendFailed = i(857), o))(et || {});
((o) => {
  const e = i, t = { vpOoO: function(a, f) {
    return a != f;
  }, WCsGh: function(a, f) {
    return a == f;
  }, fNPdS: e(1205), wRojh: function(a, f) {
    return a instanceof f;
  }, AnOqz: function(a, f) {
    return a === f;
  }, HMiEw: "IWOju", bVmuO: "imageException", uzOLn: e(450), HiCYO: "imageToGray4Failed", VgzNE: "sendFailed", JTeyG: "success", OzYmu: function(a, f) {
    return a == f;
  }, RSCqe: "number", BsnuX: function(a, f) {
    return a(f);
  }, YCYKo: function(a, f) {
    return a(f);
  }, FiaIF: "imageexception", OkGAj: "imagesizeinvalid", ILULc: e(1040), GCIoG: e(594), rldbJ: e(985), OUzXo: e(680), pZXKj: e(1135), OulOk: function(a, f) {
    return a == f;
  }, clOsn: e(487), JXIXv: function(a, f) {
    return a(f);
  }, oxEeG: function(a, f) {
    return a === f;
  }, ZLfhi: function(a, f) {
    return a(f);
  }, tyWJO: "invalid", UmgLv: function(a, f) {
    return a === f;
  }, IQoRP: "heDLR", mgxfL: function(a, f) {
    return a === f;
  }, NVgVN: function(a, f) {
    return a === f;
  } };
  function n(a) {
    const f = e;
    if (t.AnOqz(t[f(937)], t[f(937)])) switch (a) {
      case f(1067):
        return 0;
      case t[f(794)]:
        return 1;
      case t[f(660)]:
        return 2;
      case t[f(379)]:
        return 2;
      case t[f(919)]:
        return 3;
      default:
        return 3;
    }
    else if (t.vpOoO(_0x385828, null)) {
      if (t[f(467)](typeof _0x3e2c51, t.fNPdS)) return _0x392788;
      if (_0x44f28d.isArray(_0x255ae2)) return _0x3cd901[f(949)]((w) => _0x65a397(w) & 255);
      if (_0x1ff366 instanceof _0x3086f0) return _0x46f162[f(439)](_0x4ee401);
      if (t.wRojh(_0x1a2c29, _0x543eb5)) return _0x4e2871[f(439)](new _0xe6f93b(_0x48cf56));
    }
  }
  o[e(1003)] = n;
  function x(a) {
    const f = e;
    switch (a) {
      case 0:
        return t[f(938)];
      case 1:
        return t[f(794)];
      case 2:
        return t[f(660)];
      case 3:
        return t[f(919)];
      default:
        return t[f(919)];
    }
  }
  o[e(355)] = x;
  function r(a) {
    const f = e;
    if (t[f(912)](typeof a, t[f(1131)]) && Number[f(548)](a)) return t[f(537)](x, a);
    if (t[f(467)](typeof a, t[f(471)])) {
      let w = t[f(832)](Number, a);
      if (Number[f(548)](w)) return t.YCYKo(x, w);
      let b = a[f(933)]().toLowerCase();
      switch (b[f(511)](".") ? b.split(".")[f(1010)]() : b) {
        case t[f(938)]:
          return t[f(938)];
        case t[f(466)]:
          return t[f(794)];
        case t.OkGAj:
          return t[f(660)];
        case t[f(1175)]:
        case t.GCIoG:
        case t.rldbJ:
        case t.OUzXo:
          return t[f(379)];
        case t.pZXKj:
          return t.VgzNE;
        default:
          return t[f(919)];
      }
    }
    if (a && t[f(732)](typeof a, t[f(631)])) {
      if (t[f(912)](typeof a[f(792)], t.RSCqe)) return t[f(832)](x, a.index);
      if (typeof a[f(862)] == t.RSCqe) return t[f(664)](x, a[f(862)]);
      if (t[f(732)](typeof a.name, t[f(471)])) return t[f(537)](r, a[f(565)]);
    }
    return t[f(919)];
  }
  o.normalize = r;
  function u(a) {
    const f = e;
    return t.oxEeG(a, t[f(938)]);
  }
  o[e(695)] = u;
  function s(a) {
    const f = e;
    if (t[f(431)](f(402), t[f(856)])) return t[f(1186)](a, t[f(794)]);
    {
      let w = t[f(766)](_0x7ed957, _0x50b1c6);
      if (_0x4322ed[f(548)](w)) return t.JXIXv(_0x171176, w);
      let b = _0x5183f4[f(933)]()[f(939)]();
      switch (b[f(511)](".") ? b[f(672)](".").pop() : b) {
        case f(1067):
          return 0;
        case t[f(404)]:
          return 1;
        case f(1062):
          return 2;
        case f(942):
        case f(1051):
          return 3;
        default:
          return 1;
      }
    }
  }
  o[e(962)] = s;
  function c(a) {
    const f = e;
    return t[f(682)](a, t[f(660)]);
  }
  o[e(1015)] = c;
  function l(a) {
    return a === t.HiCYO;
  }
  o[e(351)] = l;
  function d(a) {
    const f = e;
    return t[f(1186)](a, t[f(919)]);
  }
  o[e(772)] = d;
})(et || (et = {}));
var zt = class F {
  constructor(e = {}) {
    Object.assign(this, e);
  }
  static fromJson(e) {
    const t = i, n = { MaekH: function(f, w, b, z) {
      return f(w, b, z);
    }, ehkor: "ContainerTotalNum", ixezk: function(f, w) {
      return f != w;
    }, LIeIe: function(f, w, b, z, K) {
      return f(w, b, z, K);
    }, fLCas: "listObject", nDIPN: t(1200), iaDeT: "List_Object", GnkXP: function(f, w) {
      return f != w;
    }, BvMGd: t(360), jxWsw: "Text_Object", WEeQP: function(f, w, b, z, K) {
      return f(w, b, z, K);
    }, WyaFC: t(562), jumlz: "ImageObject", UUgQv: "Image_Object", MpnXf: "widgetId", tZYTj: t(995), ZvCwe: t(718), LstiS: function(f, w, b) {
      return f(w, b);
    } };
    var x, r, u;
    let s = n.MaekH(v, e, t(842), n[t(1048)]), c = (n[t(844)](x = n[t(559)](v, e, n[t(668)], n[t(399)], n[t(983)]), null) ? x : []).map(O[t(916)]), l = (n.GnkXP(r = v(e, n[t(837)], "TextObject", n[t(587)]), null) ? r : [])[t(949)](X[t(916)]), d = (n[t(784)](u = n[t(604)](v, e, n.WyaFC, n.jumlz, n[t(1022)]), null) ? u : []).map(k[t(916)]), a = v(e, n[t(796)], n[t(389)], n[t(931)]);
    return new F(n.LstiS(A, D({}, e ?? {}), { containerTotalNum: s, listObject: c, textObject: l, imageObject: d, widgetId: a }));
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { YnKHj: function(x, r) {
      return x instanceof r;
    } }[t(1192)](e, F) ? e[t(1128)]() : new F(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { ntDCM: function(u, s, c) {
      return u(s, c);
    }, rCGzl: function(u, s) {
      return u != s;
    }, OmBaW: function(u, s) {
      return u == s;
    }, IDpXR: function(u, s) {
      return u == s;
    }, ykori: function(u, s) {
      return u == s;
    }, agdSc: function(u, s) {
      return u == s;
    } };
    var n, x, r;
    return t[e(771)](A, D({}, t[e(1183)](this, null) ? this : {}), { containerTotalNum: t.OmBaW(this, null) ? void 0 : this[e(842)], listObject: t[e(1116)](n = t.OmBaW(this, null) ? void 0 : this[e(797)], null) ? void 0 : n[e(949)](O[e(1128)]), textObject: t[e(412)](x = t[e(412)](this, null) ? void 0 : this[e(360)], null) ? void 0 : x.map(X.toJson), imageObject: t.agdSc(r = t[e(412)](this, null) ? void 0 : this.imageObject, null) ? void 0 : r[e(949)](k[e(1128)]), widgetId: t.IDpXR(this, null) ? void 0 : this[e(1115)] });
  }
}, mt = class Z {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { jhtBB: function(a, f, w, b) {
      return a(f, w, b);
    }, VAigv: "containerTotalNum", zVugD: t(397), SqfSY: function(a, f, w, b, z) {
      return a(f, w, b, z);
    }, MnOkg: t(797), XSoGh: "ListObject", byBrY: t(360), aRpkl: "TextObject", KZzBj: t(911), yMrkD: "Image_Object", apVqv: function(a, f, w) {
      return a(f, w);
    }, WcirA: function(a, f) {
      return a != f;
    } };
    var x, r, u;
    let s = n[t(361)](v, e, n.VAigv, n[t(633)]), c = ((x = n[t(583)](v, e, n[t(671)], n.XSoGh, t(1114))) != null ? x : []).map(O.fromJson), l = ((r = v(e, n[t(350)], n[t(602)], n.KZzBj)) != null ? r : [])[t(949)](X.fromJson), d = ((u = v(e, "imageObject", t(489), n[t(877)])) != null ? u : [])[t(949)](k[t(916)]);
    return new Z(n[t(1111)](A, n[t(1111)](D, {}, n[t(619)](e, null) ? e : {}), { containerTotalNum: s, listObject: c, textObject: l, imageObject: d }));
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { PGCLH: function(x, r) {
      return x instanceof r;
    } }.PGCLH(e, Z) ? e.toJson() : new Z(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { AaIUi: function(u, s, c) {
      return u(s, c);
    }, PQRTN: function(u, s, c) {
      return u(s, c);
    }, vKnCc: function(u, s) {
      return u != s;
    }, DFIrv: function(u, s) {
      return u == s;
    }, VsRSf: function(u, s) {
      return u == s;
    }, evxVk: function(u, s) {
      return u == s;
    }, toPRU: function(u, s) {
      return u == s;
    } };
    var n, x, r;
    return t.AaIUi(A, t[e(767)](D, {}, t[e(1056)](this, null) ? this : {}), { containerTotalNum: t[e(1007)](this, null) ? void 0 : this[e(842)], listObject: t[e(514)](n = t[e(1146)](this, null) ? void 0 : this[e(797)], null) ? void 0 : n.map(O[e(1128)]), textObject: (x = t[e(982)](this, null) ? void 0 : this[e(360)]) == null ? void 0 : x[e(949)](X[e(1128)]), imageObject: (r = this == null ? void 0 : this.imageObject) == null ? void 0 : r.map(k[e(1128)]) });
  }
}, j = ((o) => (o[o[i(373)] = 0] = i(373), o[o.SCROLL_TOP_EVENT = 1] = i(655), o[o.SCROLL_BOTTOM_EVENT = 2] = i(813), o[o[i(810)] = 3] = i(810), o[o[i(590)] = 4] = "FOREGROUND_ENTER_EVENT", o[o[i(376)] = 5] = "FOREGROUND_EXIT_EVENT", o[o.ABNORMAL_EXIT_EVENT = 6] = i(1173), o[o.SYSTEM_EXIT_EVENT = 7] = i(869), o[o[i(1029)] = 8] = i(1029), o))(j || {});
((o) => {
  const e = i, t = { dAZHt: function(x, r) {
    return x == r;
  }, Nkarw: e(929), mrcFf: function(x, r) {
    return x >= r;
  }, RZiwn: function(x, r) {
    return x <= r;
  }, hKgjg: "CLICK_EVENT", CCRad: e(1132), CqgnP: e(655), aHCuN: "SCROLL_TOP", fReiM: e(1002), XHkjQ: e(810), JtHWw: e(970), czbZJ: "FOREGROUND_ENTER_EVENT", QAjrW: e(573), nfzlK: e(376), chKKE: e(1173), YYmtb: e(691), QHFdi: "SYSTEM_EXIT", fnyDB: e(1029), TtHbC: e(1148), eEKPk: e(807) };
  function n(x) {
    const r = e;
    if (t.dAZHt(typeof x, t[r(1061)]) && Number[r(548)](x)) return t[r(645)](x, 0) && t[r(1197)](x, 8) ? x : void 0;
    if (typeof x == r(1205)) switch (x[r(933)]()[r(861)]()) {
      case t[r(535)]:
      case t[r(599)]:
        return 0;
      case t[r(534)]:
      case t.aHCuN:
        return 1;
      case "SCROLL_BOTTOM_EVENT":
      case t[r(1078)]:
        return 2;
      case t[r(570)]:
      case t.JtHWw:
        return 3;
      case t.czbZJ:
      case t.QAjrW:
        return 4;
      case t[r(1140)]:
      case r(834):
        return 5;
      case t[r(560)]:
      case t[r(662)]:
        return 6;
      case "SYSTEM_EXIT_EVENT":
      case t[r(634)]:
        return 7;
      case t[r(892)]:
      case t[r(1021)]:
      case t.eEKPk:
        return 8;
      default:
        return;
    }
  }
  o.fromJson = n;
})(j || (j = {}));
var nt = ((o) => (o[o[i(640)] = 0] = i(640), o[o.TOUCH_EVENT_FROM_GLASSES_R = 1] = "TOUCH_EVENT_FROM_GLASSES_R", o[o[i(428)] = 2] = i(428), o[o.TOUCH_EVENT_FROM_GLASSES_L = 3] = i(920), o))(nt || {});
((o) => {
  const e = i, t = { IRrMj: function(x, r) {
    return x(r);
  }, HleUo: e(1067), dGFzP: "imageexception", AJjWg: e(1040), ykbbv: "imageto_gray4failed", tOCQX: e(1135), AeckG: e(408), ncFEX: function(x, r) {
    return x == r;
  }, nyQrt: e(929), aEFVn: function(x, r) {
    return x >= r;
  }, DSyzH: function(x, r) {
    return x <= r;
  }, dJMWc: e(1205), lYqyQ: e(640), lDwfC: e(1179), XwiTg: e(930), HVElY: e(592), KStGu: e(858), JGbVy: e(1071), bCOML: "GLASSES_L" };
  function n(x) {
    const r = e, u = { nNePV: function(s, c) {
      return t[L(1166)](s, c);
    }, MoKHq: t[r(433)], JGAQu: t[r(575)], vlLMy: t.AJjWg, tXeaW: t[r(406)], GXJKx: "image_to_gray4_failed", WIVAm: t.tOCQX, vxfem: r(857) };
    if (r(408) !== t[r(891)]) {
      let s = _0x442b21(_0x647223);
      if (_0x509d43[r(548)](s)) return u[r(579)](_0xa7d5af, s);
      let c = _0x210758[r(933)]()[r(939)]();
      switch (c.includes(".") ? c[r(672)](".")[r(1010)]() : c) {
        case u[r(1047)]:
          return u[r(1047)];
        case u.JGAQu:
          return r(1169);
        case r(1125):
          return "imageSizeInvalid";
        case u.vlLMy:
        case r(594):
        case u[r(1064)]:
        case u[r(823)]:
          return r(936);
        case u.WIVAm:
          return u.vxfem;
        default:
          return u[r(752)];
      }
    } else {
      if (t.ncFEX(typeof x, t[r(387)]) && Number[r(548)](x)) return t[r(778)](x, 0) && t[r(782)](x, 3) ? x : void 0;
      if (typeof x == t[r(839)]) switch (x[r(933)]()[r(861)]()) {
        case t[r(413)]:
        case t[r(692)]:
        case t[r(385)]:
          return 0;
        case t.HVElY:
        case t[r(637)]:
          return 1;
        case r(428):
        case t.JGbVy:
          return 2;
        case r(920):
        case t[r(607)]:
          return 3;
        default:
          return;
      }
    }
  }
  o[e(916)] = n;
})(nt || (nt = {}));
function rt() {
  const o = ["BgLZDe9IAMvJDa", "rNbgqKG", "qKvdrwy", "v3HoyxK", "BeL6A0u", "zKzZr0u", "q29UBMvJDgvK", "udGWma", "EvHSzui", "txPmtK8", "su1vx1jfue9sva", "CgnT", "q29UBMvJDgLVBKzHAwXLza", "re9vqKXfx0nmsunlx0vwru5u", "t1f0A04", "qM9YzgvYx0nVBg9Y", "u0nst0Xmx0jpvfrptv9fvKvova", "AMLLwLu", "whDpCMe", "Bw9KzwW", "sxnfDMvUDenHChr1CMu", "tvbTD1y", "AvLqwwq", "wxvXqvi", "r1rgrMq", "Defgrgq", "r1Hks3G", "mtaXnZflq2LpvKi", "C3rHDhvZ", "zwPirxa", "C1Dgt04", "z1ruBMe", "ufzpr0G", "AxnhBgfZC2vZ", "quHHweq", "wunzs28", "EwjWAgi", "rK9sruDst1vorf9fweLu", "wuLQtMW", "ugfKzgLUz0XLBMD0Aa", "qNznr2q", "vMH5DvK", "zePnv2m", "wvrkC0u", "y2LvqLy", "y29UDgfPBMvYvg90ywXoDw0", "qM9YzgvYx1jHzgL1CW", "AxHLEMS", "v2HJzvO", "wKT0zwy", "DfPcDey", "zxzLBKfWCeXHDw5JAfnVDxjJzq", "veX3tva", "wf9qB3nPDgLVBG", "Aw52ywXPza", "sePJruK", "DhLWzq", "DgzHtxG", "Afr3yu8", "svfVuLa", "C2vUzezHAwXLza", "r0Xbu1nfu19s", "D2vfteq", "ANjUz28", "Dg9vChbLCKnHC2u", "DMfSDwu", "B2TAvwO", "y3jLyxrLrgvMyxvSDa", "Cw5buwS", "su1LuxG", "twfWuMf3rgf0yq", "BgTWAhq", "u1Ltvevnx0vysvrFrvzftLq", "q29UDgfPBMvYsuq", "yvj1y3m", "vg1Ptxu", "t05cq20", "y3DnCxi", "wNDethu", "wuvTCKK", "Eu1YA0q", "ANnVBKrHDge", "yvntELi", "yuHsy1G", "u1f3u1e", "wfHQv24", "B25ezxzPy2vtDgf0DxndAgfUz2vK", "zgLZCgf0y2HszwfKEuv2zw50", "D1DNDvu", "C1zivwS", "yMz3quq", "s2LpChu", "mJCWzNLADvjs", "zMvuzeq", "qwvJA0C", "zM55rei", "BwfWuMf3rgf0yq", "uMzdz1a", "qM9YzgvYuMfKAxvZ", "ueDTD3i", "Bxv5wfe", "EgrLELu", "zfzPvLG", "CfHRu0u", "y29TCgXLDgu", "Cg9ZDe1LC3nHz2u", "rhjgCuO", "twfWx1jHD19eyxrH", "u21TB3y", "AxrLBunVBNrHAw5LCG", "zNvUy3rPB24", "mti0nJHgvhv0sMC", "t3rhDuK", "yxbWtwvUDq", "vgv4Df9pyMPLy3q", "t3PzBxu", "seTjyuS", "u1bcwuy", "t2jJAgO", "zNjVBuPZB24", "u2TOCw8", "qufqz24", "vMD6tKu", "ve9vq0HFrvzftLrFrLjptv9hteftu0vtx0W", "C2LlEeC", "CgLwCha", "wMj5q24", "B2zZwMy", "BgLZDgvUx2v2zw5FyxbWx2rHDge", "vK9Yvhe", "rhjwz2m", "twfWrNjHz21LBNrqywnRzxrtAxPL", "BNvTyMvY", "rfvntvLFtLvmta", "wNzdD2u", "twfWx0zYywDTzw50x0LUzgv4", "DhjPBq", "yxbWBhK", "udyWma", "Aw1Hz2vuB0DYyxK0rMfPBgvK", "se1PrxC", "sLrLEuC", "Dg9mB3DLCKnHC2u", "qvz0AeO", "y29UBMvJDgvK", "B3v0B2zTzw1VCNK", "rNDADKm", "DuXbuMm", "A1b5ywC", "y2fSBeHHBMrSzxi", "twfWrNjHz21LBNrjBMrLEa", "vfPhvwC", "BwfW", "sxnjDgvTu2vSzwn0qM9YzgvYrw4", "AxndB25Uzwn0Aw9UrMfPBgvK", "zxfHrvi", "CMvTB3zLrxzLBNrmAxn0zw5LCG", "udqWma", "AgfUzgXLr2XHC3nLC1n0yxr1C0nOyw5Nzwq", "B01bBue", "wu13wwG", "Aw50zxjHy3rPDMu", "DxnmyMO", "uwnHvuq", "CxHetge", "AxnjBwfNzuv4y2vWDgLVBG", "Dwzbsxu", "zhrXBKC", "sM5Ny2W", "sMHbtfO", "EvvvrM0", "z2Lks2K", "tuXYBLq", "re9vqKXfx0nmsunl", "y29TChjLC3nnB2rL", "wMTqAhG", "ywzSBKW", "y29UBMvJDfr5Cgu", "DufVsu4", "sNrAr3C", "AhnSDw0", "yM9YzgvYv2LKDgG", "zNb5qLC", "A2zTsxC", "zuHWCe4", "Dg9quLu", "Awfezvq", "sxrLBv9dB250ywLUzxi", "Aw1Hz2v0B19NCMf5ngzHAwXLza", "DMnfwwi", "Aw11uMvWB3j0rw4", "AfHSEg4", "rNDrtgy", "EfvYCLO", "y0viz3C", "ugLXuhG", "su1vrgf0yq", "z2riwhu", "v2LKz2v0swq", "qKfQAMG", "uNzgquy", "rxzLBKfWCejYAwrNzq", "rxzLBNrFvhLWzq", "ChjVCgvYDhLjC0vUDw1LCMfIBgu", "qvbqx1jfuvvfu1rFuKvcvuLmrf9qquDfx1nvq0nfu1m", "u0nst0Xmx0jpvfrptq", "DMfSDwvdB2rL", "Dw5KzwzPBMvK", "tLvLrgG", "C3rYAw5NAwz5", "rezjCNy", "sgLTqNC", "CgfYC2u", "Cg9W", "AvPWELu", "tfHVtMm", "qvbqx1jfuvvfu1rFvvbhuKferv9jtufhrv9sqvDFrefuqv9gquLmruq", "uurdv2W", "AxnjBwfNzvnPEMvjBNzHBgLK", "vwnfwgi", "twfWx1nLC3nPB25Fswq", "u3LZDgvTrxHPDfjLyxnVBKnVzgu", "twfWvg90ywXtAxPL", "zhPyEMO", "vhriyKm", "vvvNuxy", "rNnUCue", "DfvstKS", "yxzHDgfY", "r2v0vxnLCKLUzM8", "EeTUtgK", "uuTAuMq", "su1vx0rbvefFuKvqt1ju", "AgvPz2H0", "BM9Uzq", "z2v0t3DUuhjVCgvYDhLtEw1IB2XZ", "BKvgqw4", "z2v0tg9JywXtDg9YywDL", "C3LZrxzLBNq", "wgvMq1m", "su1vx1jLCg9YDevU", "seLdAeW", "r0Lfs3a", "Aw1Hz2v0B2DYyxK0zMfPBgvK", "yxvKAw9fDMvUDa", "teHPsue", "CMLHvuG", "suPkAeS", "rgTqCMG", "uxD3ugO", "tw9lshe", "zwHRB3i", "wK1tvhK", "uwPhDvO", "B3v0x29Mx21LBw9YEq", "v0DnDfe", "qvbqx1jfuvvfu1rFq1jfqvrfx09vve9gtuvnt1jzx0nptLrbsu5fuG", "B2zbsgm", "C2r1s2S", "DKTUq2m", "DeXxuxK", "yuTUvK8", "zMX1DhrLCL9PBMfWChDLyNzPzxC", "q1vby04", "tMTHCNC", "B3zLCNnPEMu", "z2v0sw5ZDgfUy2u", "DfHLyvC", "DwvbDgC", "wgjbEhy", "C3vJy2vZCW", "wgnxr1u", "qxvKAw9dB250CM9S", "qNL6swu", "uKLorW", "EuHNyNi", "q29UDgvUDe9MzNnLDa", "AxnjBKnHC2u", "EevkBwq", "u1DStfC", "yLz2Exi", "zLjLAu0", "y0nWtfa", "z2v0vxnLCKLUzM8", "AgzSDfi", "AxnsAw5N", "CNbusuK", "mJmYrMDZz1jx", "yujOvgK", "q29UDgfPBMvYtMfTzq", "D2fYBG", "tNfxtfq", "txnove8", "C3LZDgvTrxHPDfjLyxnVBKnVzgu", "Au1vrgf0yq", "nJCWmti0vgfmCgXh", "weT4uKW", "q29UDgfPBMvYx05HBwu", "rwnSuMe", "u3nLDfm", "r1vbwvO", "AwnPDhq", "DvrWzMS", "tfzUrg4", "rML0CKC", "DKrwt0G", "ze5druC", "CefNA3K", "yM9YzgvYq29SB3i", "uMLUzZe", "uvPdAKe", "sK1lyNq", "v2Pgww4", "tLD3yMy", "yxbwCxy", "BgLZDa", "s1nPB0e", "tgLZDf9pyMPLy3q", "D2LKz2v0swq", "surWwfi", "Eerctwi", "tgLPzLC", "zg9Uzq", "tgPWwum", "t1vWvMC", "DMTbwuG", "rwzbwKC", "EKjcrMO", "Aw1Hz2vZAxPLAw52ywXPza", "CMvZB2X2zq", "vNbUu08", "Dg9kC29U", "AxnjDgvTu2vSzwn0qM9YzgvYrw4", "C2DnBgm", "uLndCwu", "q0Xjq0S", "A2HUz1y", "q2LQB2u", "C2vUzgzHAwXLza", "Efb5zLm", "v1bwrvO", "uuHhExe", "yLfcDhK", "BMz6BeS", "y3jLyxrLu3rHCNrvCfbHz2vdB250ywLUzxi", "yMDJqw0", "u1fgq0u", "thDlEge", "wv9qB3nPDgLVBG", "zxz4vMS", "zwj2D0O", "su1vx0rbvee", "zgvMAw5LuhjVCgvYDgLLCW", "ELvgyKK", "twfWu2vZC2LVBKLK", "z2XHC3nLC01LBNu", "CgfKzgLUz0XLBMD0Aa", "ELfwsLK", "y29UDgvUDeXLBMD0Aa", "y29UDgfPBMvYsuq", "A1HwzhC", "D2LUzg93igLZig5VDcbKzwzPBMvK", "AgDNzue", "DLfus20", "ruTiCKu", "sgvPz2H0", "udeWmda", "ugfKzgLUz19mzw5NDgG", "uMvWB3j0rNjX", "svjYtwO", "r1vHwui", "BhfXwfq", "Aw1Hz2vfEgnLChrPB24", "CNjHDuq", "ChjVDg90ExbL", "ufbnrLC", "qujot1jnquXFrvHjvf9fvKvova", "wfbVC2L0Aw9U", "suXvtgm", "r1zgvha", "qufpEw8", "qvbqx1jfuvvfu1rFqvvesu9Fq1rsx0zbsuXfra", "ve9vq0HFrvzftLrFrLjptv9evu1nwv9ovuXm", "vLv0Awe", "zwzPufm", "q3nVs2C", "CKnhEMW", "s01PBNq", "t29ituW", "BwD4zKW", "B1rXAui", "rgLZy29UBMvJDgvK", "y0XPu1e", "u2v0tg9JywXtDg9YywDL", "C0fgrfi", "ww5lsgO", "qvbqx1jfuvvfu1rFvvbhuKferv9irufsvejfqvrFuefds0vux1nvq0nfu1m", "vxbKyxrLsw1Hz2vsyxDeyxrH", "qKH3CuO", "C2v0tg9JywXtDg9YywDL", "uLPPD24", "Dgv4DenVBNrHAw5LCLvWz3jHzgu", "y2fSBf9LDMvUx2fWCf9TzxrOB2q", "tgLZDe9IAMvJDa", "qvbqx1jfuvvfu1rFq1jfqvrfx0LovKfjterFq09ovefjtKvs", "DMfQv04", "zgv0ywLS", "rxfMD1O", "C3rYAw5N", "wur4zKC", "EezHsMi", "Aw5ZDgfUy2u", "yNLcCLK", "AxnjBwfNzvrVr3jHEtrgywLSzwq", "w0v2zw5bChbcCMLKz2vDihbVC3rnzxnZywDLoIbgBhv0DgvYigHHBMrSzxiGBM90igf2ywLSywjSzq", "CMzyvui", "AxndB25Uzwn0Aw5N", "zNjVBuLUDa", "BLDTAMu", "v0vyELC", "EfrArve", "C0jnwhK", "Dgv4De9IAMvJDa", "AMH0qKi", "zhPewwq", "qM9YzgvYx1DPzhrO", "z2v0rgv2AwnLsw5MBW", "uNzuwMu", "ANHWuvC", "BNvdqK0", "v0nUEey", "CMvWB3j0rNjX", "twfWx1rVDgfSx1nPEMu", "thrPtee", "y29UDgfPBMvYtMfTzq", "q0Xjq0TFrvzftLq", "zxPrzNu", "mJy2oduZvwHPt3Ds", "rK9sruDst1vorf9fweLux0vwru5u", "AxnfDMvUDenHChr1CMu", "EfbVC2L0Aw9U", "sgLdwu8", "sgreveq", "q1LlqLm", "DfzQyxu", "zgLZCgf0y2HfDMvUDa", "CMLUzZe", "whDPvgC", "BwfWrNjHz21LBNrqywnRzxrtAxPL", "BNLrCNq", "y29UBMvJDgLVBKzHAwXLza", "DfPzvgO", "CMvIDwLSzfbHz2vdB250ywLUzxi", "mta2nJyYmhbxB2zZsW", "r2PKveK", "DuDJBxm", "wKLWsKu", "B3v0t2znzw1VCNK", "qvbqx1jfuvvfu1rFq1jfqvrfx09wrvjtsvPfx1jfu1bptLnfx0nptLrbsu5fuG", "q29UDgfPBMvYvg90ywXoDw0", "EuPNs2G", "BKrjue4", "EgTVteu", "DxLVsuG", "Agvetfi", "ugTAwwW", "DhLxsK8", "tM9Uzq", "EwTIyNy", "AxndAgfYz2LUzW", "rLjfBfO", "DxbKyxrLu3rHDhvZ", "uwzdwe4", "vuj6uwq", "EwTVCMK", "BfLXEve", "su1vx1jLCg9YDevUx1n0yxr1CW", "CLLPBg0", "re9nq29UDgvUDeXVywrLza", "zg9KBuq", "ywTftee", "AKfVww8", "AxrLBu5HBwu", "y0z3A2G", "rxHkEMe", "sKHjv3i", "Aw11uMvWB3j0rw5tDgf0Dxm", "s2jVBfu", "CKfMyvu", "qvbqx1jfuvvfu1rFvvbhuKferv9tsfvure9xtL9gquLmruq", "ve9vq0HFrvzftLrFrLjptv9ssu5h", "wK50Dfq", "u2H1DerVD25qywDLq29UDgfPBMvY", "vw1Nthy", "yMjwBMy", "sgXLvw8", "zgvMAw5LuhjVCgvYDhK", "qM9YzgvYq29SB3i", "vLPTugy", "tNnps1G", "AxnoB25L", "zNjVBq", "y0Xyswq", "z3jcwM0", "BhvYzgq", "uxDRtxO", "v2LKDgG", "sxrLBv9xAwr0Aa", "qufWuvi", "tuDJAeG", "y29UDgvUDa", "reTLC1y", "Aw1Hz2vtAxPLsw52ywXPza", "ueTJs00", "Eu55suC", "DeXLy3e", "EvbVC2L0Aw9U", "v0vlzxO", "zKDxCei", "tezewgO", "z2v0r2XHC3nLC0LUzM8", "EeDOBvm", "DfvduvG", "su1vuMvWB3j0rw4", "sNvtthi", "qLPUC1O", "z052rgC", "ExvoA2S", "rMLHsuy", "v0nZr2G", "u1jsyMq", "qvbqx1jfuvvfu1rFqvvesu9Fq1rsx1nvq0nfu1m", "yxvKAw8", "zK5qzfm", "vwvqzgS", "sevqugO", "AhjKwu8", "qLfdsNO", "BM9YBwfSAxPL", "u2fMD1C", "yMHjqK8", "z2PMv2W", "sgreAKu", "udeWma", "txrNrNe", "uunntLq", "DgP0tuu", "y2fSBa", "tu1JtfO", "B2jQzwn0", "quzYqKq", "sw1Hz2vpyMPLy3q", "BgPuy2C", "BgLZDev2zw50", "AunUq3q", "tu9Rv0e", "yuPcEu8", "r2T3D0O", "qun0Efq", "Dez0zhi", "sxnFzxzLBNrFy2fWDhvYzq", "BNjTwKi", "AgfZt3DUuhjVCgvYDhK", "Aw11q29UDhjVBa", "B1PoB2u", "BwfWu2vZC2LVBKLK", "y3vYCMvUDfnLBgvJDeL0zw1oyw1L", "CMvHzhLtDgf0zq", "tNvytvy", "A2PSAKe", "qvbqx1jfuvvfu1rFvvbhuKferv9urvHux0rbvefFu1vdq0vtuW", "Dejuwum", "q3vYCMvUDfnLBgvJDf9jDgvTtMfTzq", "Aw5JBhvKzxm", "ruXJDhi", "BNrewg0", "vNnsu2y", "w0v2zw5bChbcCMLKz2vDifvUA25VD24GBwvZC2fNzsbTzxrOB2q6", "y1n0DxO", "qvbqx1jfuvvfu1rFq1jfqvrfx1bbr0vFu1vdq0vtuW", "zxzLBNruExbL", "vvn4CMW", "sxrLBv9dB3vUDa", "Bw5fv3q", "su1vx0rHDge", "zgrSs3e", "BwLpywi", "shvPEuK", "D2LKDgG", "AvHjsw0", "BxrHyxm", "wg1utNu", "wvPqzvu", "DMXoCui", "AgfUzgXLrxzLBKH1yKv2zw50q2HHBMDLza", "BxvmC0C", "q3fNBLa", "AeTNAMC", "yw1ryve", "qNnUDvG", "DgT5BMu", "qM9YzgvYv2LKDgG", "sxnFsxrLBv9tzwXLy3rFqM9YzgvYx0vU", "sw11q29UDhjVBa", "CxHpDfO", "DgHYB3C", "AgPmrvC", "Bg9N", "uufdz0C", "q3vYCMvUDfnLBgvJDf9jDgvTsw5KzxG", "AxngAw5PDgu", "q1rJwNO", "zw1rwvq", "tgzpEvq", "zezRzhe", "wvbVC2L0Aw9U", "mtmZoduWv0ryvNDH", "sxrLBu5HBwu", "DuPdv0G", "v0rHAKq", "BgvUz3rO", "teLLswu", "y2Hls0u", "zgLZy29UBMvJDgvK", "Aw1Hz2vpyMPLy3q", "uwDVCxu", "qvbqx1jfuvvfu1rFvvbhuKferv9urvHux0rbvefFrKfjteve", "BMfTzq", "ve1IAu0", "rgv2AwnLu3rHDhvZq2HHBMDLza", "Exj2sKm", "BffKqxG", "weHRALe", "Au1vuMvWB3j0rw4", "EhPgqxm", "rK9sruDst1vorf9ftLrfuG", "sw1Hz2vFrgf0yq", "zeDgELa", "AwPvu3K", "udCWma", "DNfxC1u", "BK5Lufy", "q3vYCMvUDfnLBgvJDeL0zw1oyw1L", "q2fSBev2zw5bChbnzxrOB2q", "CgXyy0y", "u3fMu1K", "BunSA3O", "zg5Azxa", "rxzLBKH1yKv2zw50", "ANHxC3C", "tgf1BMnOu291CMnL", "tLLZuwG", "rK9sruDst1vorf9ftLrfuL9fvKvova", "rKzKzhK", "ve9vq0HFrvzftLrFrLjptv9hteftu0vtx1i", "zxzLBNrtB3vYy2u", "Aw1Hz2v0B2DYyxLMywLSzwq", "yMf0DgvYEuXLDMvS", "uNvmC08", "tMDjzuG", "AKrLDhm", "q0nsywq", "r2v0tg9JywXtDg9YywDL", "yxvKAw9dB250CM9S", "yvjWA2W", "s2fhshO", "v0vLuva", "u1Hkr1u", "udiWma", "yKnptuW", "yvbnA3O", "qvbqx1jfuvvfu1rFvvbhuKferv9jtufhrv9sqvDFrefuqv9tvundrvnt", "vNHrEe4", "CMvHzhK", "DerQBg8", "AMv5C28", "u3LHBuq", "twTcqM8", "tgLZDgvUrxzLBKfWCe5VDgLMEq", "tKvAy1q", "Bg9hreO", "v2nPCKe", "DvHvExK", "BwfWrNjHz21LBNrjBMrLEa", "u3PbDgO", "vfzPq1O", "uMHPqNC", "zvrYC0q", "AxnbCNjHEq", "sMn3DMG", "BM90u2v0", "Bu9cvxu", "ueXHEfi", "y2XpC24", "y3vYCMvUDfnLBgvJDeL0zw1jBMrLEa", "ELz1z0q", "uuHgzgK", "Dgv4Dev2zw50", "zNjVBvn0CMLUzW", "s1n0r3u", "wejTBhy", "zLz5tMO", "ve9vq0HFrvzftLrFrK9stv9evu1nwv9ovuXm", "EvzwEMu", "AvLcBey", "sLL4AhC", "wKPyEhu", "BxjJrMy", "DxbKyxrLsw1Hz2vsyxDeyxrH", "Bwr2tNG", "vMHfuwe", "qvbqx1jfuvvfu1rFvvbhuKferv9tsfvure9xtL9tvundrvnt", "ywrKrxzLBNrmAxn0zw5LCG", "ugzxtwu", "rxzLBNruExbL", "vg1PDM8", "Aw5PDa", "u0nst0Xmx1rpuf9fvKvova", "DwLK", "q01SBve", "uvHSBwK", "BM9YBwfSAxPLsw1Hz2veyxrH", "DxPptg4", "C2H1DerVD25qywDLq29UDgfPBMvY", "wvLTDgi", "B2Porgu", "sLHjwhy", "yM9YzgvYuMfKAxvZ", "yujiA0S", "qMzQy04", "zKXdyxm", "y291BNrYEq", "EuXzq0G", "tw5pA2C", "C3bSAxq", "t1ber3O", "tejezKO", "mtrhwLz5D3q", "q29UDgfPBMvYx0Le", "AM9yBhe", "t3vWyKe", "DwTADuS", "Aw1Hz2vFDg9Fz3jHEtrFzMfPBgvK", "mJmXmJCWt0DRzuPJ", "tLzNvK4", "zxzLBKfWCejYAwrNzvjLywr5", "ALLmwvG", "BwDQuLu", "zxHPDe1Vzgu", "D2jPweq", "AMXsyMq", "rM9ZC0m", "qwf1rMe", "qujot1jnquXFrvHjva", "Ber3zKm", "v25SELC", "swLyuvO", "AxntDwnJzxnZ", "r3P5q0i", "uKnPzNm", "zgv2AwnLu3rHDhvZq2HHBMDLza", "wMPetLi", "w0v2zw5bChbcCMLKz2vDierLDMLJzsbZDgf0DxmGy2HHBMDLzdO", "u3j2ufi", "wxfuwNm", "yxvKAw9qy20", "DMrtzKG", "BwfWvg90ywXtAxPL", "C2TMEhe", "AhHKAxi", "AxnxzwfYAw5N", "q3jLyxrLu3rHCNrvCfbHz2vdB250ywLUzxi", "wMrYrgW", "EvPXvee", "ENHXrem", "zNHHEhK", "reDgvgS", "DhrLCgW", "udmWma", "quXZsxa", "v2LKz2v0x0Le", "EKTwu2S", "AgfUzgXLtgf1BMnOu291CMnL", "rNDyzw8", "Aw1Hz2veyxrH", "q29UDgvUDa", "rffzsMe", "uuTWBNK", "DfLKvfK", "svPly28", "x2XPC3rLBKv2zw5bChbnzxnZywDL", "DKXdwLK", "vLrkAMS", "t3LtAvK", "t3vSt2S", "vwfcqvO", "qxbOzxi", "vxD0y2C", "A2vksLK", "wxfYthi", "v2Hzq0u", "B25myxvUy2HtB3vYy2u", "wu9vt2S", "vLP1svu", "yuDzELO", "thD4rva", "ugL5qw0", "B09Tuva", "qKfHsuq", "rKjoDu8", "Aw11rgf0yq", "ChLnvNC", "r2nbt2i", "zuHYtfG", "DNHMzw0", "qvbqx1jfuvvfu1rFuKvcvuLmrf9qquDfx0zbsuXe", "r0PZqwC", "sgzJDvy", "z2v0t3DUuhjVCgvYDhLezxnJCMLWDg9YCW", "r2v0r2XHC3nLC0LUzM8", "EfvzsfC", "zxzLBKH1yKv2zw50", "x3jLywr5", "udKWma", "u3vnDvu", "zezYq00", "sxrLBvDPzhrO", "yKzvEKu", "wKXMAgK", "uffsve4", "y29UBMvJDgLUzW", "CLjRwvu", "wLH3AeS", "BNreq00", "Axntzw5KrMfPBgvK", "y2fSBev2zw5bCha", "q29TChjLC3nFtw9Kzq", "vM1VuLe", "y29UDgvUDe9MzNnLDa", "zgf0yq", "yuvgvM4", "w0v2zw5bChbcCMLKz2vDiejYAwrNzsbPBML0AwfSAxPLza", "wM9NALK", "suHWvgy", "rfn5EKG", "z2rHD2C", "r25Rwfa", "ugvrsvq", "svPWsLm", "Bu54rvO", "t1vssMK", "Ag1jyMi", "DenqCuq", "yxnZAwDU", "Aw5KzxG", "EgfWtxy", "yLzTDu8", "vfHTD1a", "txbUwgy"];
  return rt = function() {
    return o;
  }, rt();
}
function L(o, e) {
  o = o - 347;
  const t = rt();
  let n = t[o];
  if (L.PpVrvV === void 0) {
    var x = function(c) {
      const l = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789+/=";
      let d = "", a = "";
      for (let f = 0, w, b, z = 0; b = c.charAt(z++); ~b && (w = f % 4 ? w * 64 + b : b, f++ % 4) ? d += String.fromCharCode(255 & w >> (-2 * f & 6)) : 0)
        b = l.indexOf(b);
      for (let f = 0, w = d.length; f < w; f++)
        a += "%" + ("00" + d.charCodeAt(f).toString(16)).slice(-2);
      return decodeURIComponent(a);
    };
    L.tycvPW = x, L.PyQHWI = {}, L.PpVrvV = !0;
  }
  const r = t[0], u = o + r, s = L.PyQHWI[u];
  return s ? n = s : (n = L.tycvPW(n), L.PyQHWI[u] = n), n;
}
var St = class H {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { NYsQh: function(r, u) {
      return r(u);
    }, WhceZ: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, XwOra: t(518), jlRbd: t(652), HfcuV: t(999), nPMSx: "ContainerID", lNFFs: "Container_ID", Qgoqu: t(372), hggeA: t(1086), WGMtQ: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, tZBtF: t(504), xQtnp: t(580), XIdHI: t(632), uLGLv: t(547) };
    if (!n[t(589)](q, e)) return new H();
    let x = n[t(845)](P, e, n[t(815)], n[t(688)], n[t(755)]);
    return new H({ containerID: n.WhceZ(C, e, t(1156), n.nPMSx, n.lNFFs), containerName: n.WhceZ(ot, e, n[t(563)], n[t(1159)], t(1094)), currentSelectItemName: n.WGMtQ(ot, e, n[t(847)], n.xQtnp, t(510)), currentSelectItemIndex: n[t(1052)](C, e, n.XIdHI, "CurrentSelectItemIndex", n.uLGLv), eventType: j[t(916)](x) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { QwwPj: function(x, r) {
      return x instanceof r;
    } }[t(1046)](e, H) ? e[t(1128)]() : new H(e).toJson() : {};
  }
  [i(1128)]() {
    const e = i, t = { LwKxa: function(n, x, r) {
      return n(x, r);
    }, LHiIA: function(n, x) {
      return n != x;
    } };
    return t[e(1144)](D, {}, t[e(1042)](this, null) ? this : {});
  }
}, Wt = class N {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { vdSfH: function(r, u) {
      return r(u);
    }, nuCBM: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, ntDXm: t(518), OUpVg: t(652), mNxEZ: t(999), qZMJm: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, lQdAx: "ContainerID", PeQIT: t(676), pxXtJ: function(r, u, s, c, l) {
      return r(u, s, c, l);
    }, sNJIZ: t(372), fpyBW: "ContainerName", PVOGH: t(1094) };
    if (!n[t(704)](q, e)) return new N();
    let x = n[t(367)](P, e, n[t(513)], n[t(1121)], n[t(787)]);
    return new N({ containerID: n.qZMJm(C, e, "containerID", n[t(569)], n[t(785)]), containerName: n.pxXtJ(ot, e, n.sNJIZ, n[t(979)], n[t(829)]), eventType: j.fromJson(x) });
  }
  static [i(1128)](e) {
    const t = i;
    return e ? { HeqSW: function(x, r) {
      return x instanceof r;
    } }.HeqSW(e, N) ? e[t(1128)]() : new N(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = { DYthJ: function(t, n, x) {
      return t(n, x);
    }, oOrtr: function(t, n) {
      return t != n;
    } };
    return e.DYthJ(D, {}, e.oOrtr(this, null) ? this : {});
  }
}, It = class U {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static fromJson(e) {
    const t = i, n = { OySiY: function(x, r) {
      return x(r);
    }, iAWLP: function(x, r, u, s) {
      return x(r, u, s);
    }, JoJjj: function(x, r, u, s) {
      return x(r, u, s);
    }, cvurw: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n[t(731)](q, e) ? new U({ x: n.iAWLP(C, e, "x", "X"), y: n.JoJjj(C, e, "y", "Y"), z: n.cvurw(C, e, "z", "Z") }) : new U();
  }
  static [i(1128)](e) {
    const t = i;
    return e ? e instanceof U ? e[t(1128)]() : new U(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    return { feTdD: function(n, x, r) {
      return n(x, r);
    } }[i(890)](D, {}, this != null ? this : {});
  }
}, Ot = class T {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static fromJson(e) {
    const t = i, n = { ROqwv: function(c, l) {
      return c(l);
    }, piVpp: function(c, l, d, a, f) {
      return c(l, d, a, f);
    }, ybphb: "Event_Type", yrvJC: function(c, l, d, a) {
      return c(l, d, a);
    }, XmTNu: t(593), LFDXj: "EventSource", lkpht: t(522), vRbIS: t(1091), eqaER: function(c, l) {
      return c(l);
    }, ofsZf: t(1018) };
    if (!n.ROqwv(q, e)) return new T();
    let x = n[t(922)](P, e, t(518), t(652), n[t(833)]), r = n[t(568)](P, e, n[t(529)], n[t(457)]), u = P(e, t(748), t(993), n[t(868)], n.vRbIS), s;
    return n[t(952)](q, u) && (s = It.fromJson(u)), new T({ eventType: j[t(916)](x), eventSource: nt[t(916)](r), imuData: s, systemExitReasonCode: C(e, t(1090), n[t(924)]) });
  }
  static toJson(e) {
    const t = i;
    return e ? { bVvyr: function(x, r) {
      return x instanceof r;
    } }[t(1077)](e, T) ? e[t(1128)]() : new T(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { UBzQd: function(n, x, r) {
      return n(x, r);
    }, auxMg: function(n, x) {
      return n != x;
    } };
    return t[e(411)](D, {}, t.auxMg(this, null) ? this : {});
  }
}, Xt = ((o) => (o.listEvent = i(491), o.textEvent = i(635), o[i(1035)] = "sysEvent", o.audioEvent = "audioEvent", o.notSet = i(628), o))(Xt || {});
function _t() {
  return {};
}
function vt(o) {
  const e = i, t = { YMwYh: function(x, r) {
    return x == r;
  }, MOkWA: e(491), xapMv: e(635), ttepl: e(1035), aSSzR: "audioEvent", HdDTD: e(628), fVyNj: "text", psTrP: "sys" };
  if (t[e(957)](typeof o, e(929)) && Number[e(548)](o)) switch (o) {
    case 0:
      return t[e(493)];
    case 1:
      return t[e(793)];
    case 2:
      return t.ttepl;
    case 3:
      return t[e(879)];
    default:
      return t[e(380)];
  }
  if (typeof o != e(1205)) return e(628);
  let n = o[e(933)]()[e(939)]();
  return n ? n[e(511)](e(1112)) ? t[e(493)] : n[e(511)](t[e(639)]) ? t.xapMv : n.includes(t.psTrP) ? t[e(715)] : n[e(511)](e(470)) ? t[e(879)] : e(628) : e(628);
}
function Yt(o) {
  const e = i, t = { eKgtd: function(f, w) {
    return f === w;
  }, DYPib: function(f, w) {
    return f === w;
  }, JuSLr: e(831), aJByO: function(f, w) {
    return f(w);
  }, aflnL: function(f, w, b) {
    return f(w, b);
  }, AzTaC: function(f, w) {
    return f != w;
  }, ijUSy: function(f) {
    return f();
  }, ByzIe: function(f, w) {
    return f != w;
  }, uGcms: function(f, w) {
    return f != w;
  }, xbybP: function(f, w) {
    return f != w;
  }, JdZmr: function(f, w) {
    return f != w;
  }, kPyag: function(f, w) {
    return f(w);
  } };
  var n, x, r, u, s;
  if (Array.isArray(o))
    if (t.DYPib(t[e(462)], t.JuSLr)) {
      let [f, w] = o, b = t[e(494)](vt, f), z = t[e(494)](gt, w);
      return t[e(973)](wt, b, z);
    } else return t.eKgtd(this[e(974)], "connected");
  if (!o || t.AzTaC(typeof o, "object")) return t[e(576)](_t);
  let c = t[e(1070)](x = t[e(393)](n = o[e(853)], null) ? n : o[e(518)], null) ? x : o.name, l = t.xbybP(s = t.JdZmr(u = (r = o[e(878)]) != null ? r : o[e(777)], null) ? u : o.payload, null) ? s : {}, d = t[e(945)](vt, c), a = gt(l);
  return wt(d, a);
}
function Rt(o) {
  const e = i, t = { ezQfu: function(l, d) {
    return l != d;
  }, iCnCt: function(l, d) {
    return l == d;
  }, kTqEw: function(l, d) {
    return l == d;
  }, yHgbr: function(l, d) {
    return l != d;
  }, YlKfc: function(l, d) {
    return l == d;
  }, rMMhp: function(l, d) {
    return l == d;
  }, XXjWn: function(l, d) {
    return l != d;
  }, sAFDR: function(l, d) {
    return l != d;
  }, QCMNT: function(l, d) {
    return l instanceof d;
  }, ljTcg: e(1045), QDCWl: function(l, d) {
    return l == d;
  }, ewPmP: e(1205), AauFa: e(657), bhIBO: function(l, d) {
    return l(d);
  }, sWFON: function(l, d) {
    return l < d;
  } };
  if (t[e(1191)](o, null)) {
    if (t[e(483)](o, Uint8Array)) return o;
    if (Array[e(626)](o))
      if (t[e(490)] !== t.ljTcg) this[e(760)] = !1, this[e(654)]();
      else {
        let l = o;
        return new Uint8Array(l[e(949)]((d) => Number(d) & 255));
      }
    if (t[e(1014)](typeof o, t.ewPmP)) try {
      if (t[e(690)] !== t[e(690)]) {
        var n, x, r, u, s, c;
        return new _0x1a0de1({ sn: t.ezQfu(n = t[e(492)](_0x58d76a, null) ? void 0 : _0x9e7d07.sn, null) ? n : "", connectType: _0x583000[e(636)](t[e(374)](x = t.kTqEw(_0x50e12d, null) ? void 0 : _0x5b82a9.connectType, null) ? x : ""), isWearing: t[e(1072)](r = t.YlKfc(_0x2b9157, null) ? void 0 : _0x525193.isWearing, null) ? r : !1, batteryLevel: t[e(374)](u = t.rMMhp(_0x5a5935, null) ? void 0 : _0x5aceca[e(595)], null) ? u : 0, isCharging: t[e(882)](s = _0x513820 == null ? void 0 : _0x3fe41a[e(407)], null) ? s : !1, isInCase: t[e(882)](c = t.iCnCt(_0x2a619d, null) ? void 0 : _0x34d6b8[e(1074)], null) ? c : !1 });
      } else {
        let l = t[e(478)](atob, o), d = new Uint8Array(l[e(558)]);
        for (let a = 0; t[e(827)](a, l[e(558)]); a++) d[a] = l.charCodeAt(a);
        return d;
      }
    } catch {
      return;
    }
  }
}
function wt(o, e) {
  const t = i, n = { OtGuI: t(491), BZnsZ: t(635), hiTQi: t(1035), YqrLr: t(1041), ueAtg: function(s, c) {
    return s === c;
  }, FwQLf: t(441), ONBCm: function(s, c) {
    return s != c;
  } };
  var x, r;
  let u = { jsonData: e };
  switch (o) {
    case n[t(909)]:
      u[t(491)] = St.fromJson(e);
      break;
    case n[t(463)]:
      u.textEvent = Wt[t(916)](e);
      break;
    case n.hiTQi:
      u[t(1035)] = Ot[t(916)](e);
      break;
    case n[t(737)]:
      if (n[t(1065)](n.FwQLf, n[t(989)])) {
        let s = Rt(n[t(873)](r = (x = e[t(703)]) != null ? x : e.audio_pcm, null) ? r : e[t(808)]);
        n[t(873)](s, null) && (u.audioEvent = { audioPcm: s });
        break;
      } else _0x5d29f7.sn === this.sn && (this[t(825)] = _0x281223);
  }
  return u;
}
var yt = class V {
  constructor(e = {}) {
    Object[i(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { PGmwr: function(x, r, u) {
      return x(r, u);
    }, Uwtcg: function(x, r) {
      return x != r;
    }, BHwqJ: function(x, r) {
      return x(r);
    }, OoHML: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, efiPS: t(870), itxeG: t(676), ksmXk: function(x, r, u, s, c) {
      return x(r, u, s, c);
    }, xkoLE: t(1094), GkwwJ: function(x, r) {
      return x(r);
    }, tAFDd: function(x, r, u, s) {
      return x(r, u, s);
    }, MGchH: t(776), gDPJj: t(1073), lVIzf: t(1155), aGYzZ: "ContentLength", NAdBT: function(x, r) {
      return x(r);
    }, hxdir: t(723) };
    return new V(n[t(896)](A, D({}, n[t(735)](e, null) ? e : {}), { containerID: n[t(1195)](h, n[t(1185)](v, e, t(1156), n[t(1181)], n.itxeG)), containerName: n[t(1195)](p, n.ksmXk(v, e, t(372), "ContainerName", n[t(400)])), contentOffset: n[t(495)](h, n[t(822)](v, e, n[t(447)], n.gDPJj)), contentLength: h(n[t(822)](v, e, n.lVIzf, n[t(742)])), content: n.NAdBT(p, n.tAFDd(v, e, t(448), n[t(707)])) }));
  }
  static toJson(e) {
    const t = i;
    return e ? { WnRQk: function(x, r) {
      return x instanceof r;
    } }.WnRQk(e, V) ? e[t(1128)]() : new V(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    return D({}, { vqWsU: function(n, x) {
      return n != x;
    } }[i(578)](this, null) ? this : {});
  }
}, ht = class Q {
  constructor(e = {}) {
    const t = i;
    var n;
    this[t(686)] = (n = e[t(686)]) != null ? n : 0, Object[t(791)](this, e);
  }
  static [i(916)](e) {
    const t = i, n = { UePdk: function(x, r, u) {
      return x(r, u);
    }, WPVEZ: function(x, r) {
      return x != r;
    } };
    return new Q(n[t(472)](D, {}, n[t(1137)](e, null) ? e : {}));
  }
  static [i(1128)](e) {
    const t = i;
    return e ? e instanceof Q ? e[t(1128)]() : new Q(e)[t(1128)]() : { exitMode: 0 };
  }
  [i(1128)]() {
    const e = i, t = { zQVJY: function(n, x, r) {
      return n(x, r);
    }, ELctr: function(n, x) {
      return n != x;
    } };
    return t[e(1154)](D, {}, t[e(512)](this, null) ? this : { exitMode: 0 });
  }
}, bt = class S {
  constructor(e = {}) {
    Object.assign(this, e);
  }
  static [i(916)](e) {
    const t = i, n = { vLCZY: function(x, r) {
      return x(r);
    }, GjdTI: function(x, r, u, s, c, l) {
      return x(r, u, s, c, l);
    }, HIChL: t(571), xPYyM: t(461), OQtkN: function(x, r, u, s) {
      return x(r, u, s);
    }, ByCYn: t(1165) };
    return n[t(729)](q, e) ? new S({ iMUReportEn: n[t(392)](C, e, n[t(1038)], n.xPYyM, "IMU_ReportEn", t(987)), reportFrq: n[t(811)](C, e, t(369), n.ByCYn) }) : new S();
  }
  static [i(1128)](e) {
    return e ? { plXcF: function(x, r) {
      return x instanceof r;
    } }[i(582)](e, S) ? e.toJson() : new S(e).toJson() : {};
  }
  [i(1128)]() {
    const e = i, t = { QwkMz: function(n, x, r) {
      return n(x, r);
    }, XbAxv: function(n, x) {
      return n != x;
    } };
    return t[e(443)](D, {}, t[e(1066)](this, null) ? this : {});
  }
};
(class W {
  constructor(e = {}) {
    Object.assign(this, e);
  }
  static [i(916)](e) {
    const t = i, n = { aRucs: function(x, r, u, s, c, l) {
      return x(r, u, s, c, l);
    }, vcEYb: "iMUReportEnStatus", uLARc: "IMUReportEnStatus", miOab: t(414), PPMFW: t(424) };
    return q(e) ? new W({ iMUReportEnStatus: n[t(871)](C, e, n[t(986)], n[t(944)], n[t(524)], n[t(1172)]) }) : new W();
  }
  static [i(1128)](e) {
    const t = i;
    return e ? e instanceof W ? e[t(1128)]() : new W(e)[t(1128)]() : {};
  }
  [i(1128)]() {
    const e = i, t = { rpTII: function(n, x, r) {
      return n(x, r);
    }, ZXwhK: function(n, x) {
      return n != x;
    } };
    return t[e(1083)](D, {}, t[e(770)](this, null) ? this : {});
  }
});
var Jt = class I {
  constructor() {
    const e = i;
    this[e(760)] = !1, this[e(654)]();
  }
  static getInstance() {
    const e = i;
    return I[e(349)] || (I[e(349)] = new I()), I.instance;
  }
  [i(654)]() {
    const e = i, t = { kjljA: function(n, x) {
      return n == x;
    }, IZKco: e(1205), VmoRQ: function(n, x) {
      return n === x;
    }, xUYHW: function(n, x) {
      return n || x;
    }, GTFFd: e(698), WjFYn: "evenHubEvent", ALsIp: function(n, x) {
      return n || x;
    }, hrdYO: e(515), cEHgw: function(n, x) {
      return n != x;
    }, eHppN: function(n, x) {
      return n === x;
    }, KiOpu: e(901) };
    t[e(991)](typeof window, e(1004)) && (window[e(998)] = this, t[e(981)](document[e(505)], t[e(888)]) || t[e(981)](document.readyState, e(958)) ? (this[e(760)] = !0, this.dispatchReadyEvent()) : document[e(650)](e(416), () => {
      const n = e;
      this[n(760)] = !0, this.dispatchReadyEvent();
    }), window[e(728)] = (n) => {
      const x = e;
      let r = t[x(507)](typeof n, t[x(727)]) ? JSON[x(1009)](n) : n, { method: u, payload: s, data: c } = r;
      t[x(775)](u, x(848)) ? this[x(720)](t[x(758)](c, s)) : t[x(775)](u, t[x(821)]) ? this.handleGlassesStatusChanged(t[x(758)](c, s)) : u === t[x(1109)] ? this[x(532)](t[x(717)](c, s)) : console.warn(t[x(474)], u);
    }, console[e(545)](e(779)));
  }
  [i(884)]() {
    const e = i, t = { Cijoe: function(n, x) {
      return n != x;
    }, vkAYH: function(n, x) {
      return n == x;
    }, MVPUg: e(907), xTSJL: "evenAppBridgeReady" };
    t[e(1134)](typeof window, e(1004)) && t[e(1122)](typeof window[e(383)], t.MVPUg) && window[e(383)](new Event(t.xTSJL));
  }
  get [i(611)]() {
    return this[i(760)];
  }
  [i(902)](e) {
    const t = i, n = { CUAcN: "undefined", SsetS: "evenAppMessage", kXVdw: t(352) };
    if (typeof window == n[t(1060)]) return Promise[t(1126)](void 0);
    let x = window[t(1059)];
    return x && x.callHandler ? x[t(946)](n[t(1096)], JSON[t(1006)](e)) : (console[t(1087)](n[t(1157)]), Promise[t(1126)](void 0));
  }
  [i(720)](e) {
    const t = i, n = { bNYCk: function(u, s) {
      return u != s;
    }, mnEWt: function(u, s) {
      return u == s;
    }, vwRLj: t(487), riaUH: function(u, s) {
      return u === s;
    }, FsnqA: t(910), XKxRL: function(u, s) {
      return u === s;
    }, oTqiB: t(1152), dFrCM: function(u, s) {
      return u != s;
    }, NgIeH: function(u, s) {
      return u == s;
    }, xdezU: t(848) };
    let x = n.bNYCk(e, null) && n[t(521)](typeof e, n.vwRLj) ? e.launchSource : e, r = n[t(1043)](x, n.FsnqA) || n[t(1093)](x, n[t(1187)]) ? x : n[t(1023)];
    n[t(763)](typeof window, t(1004)) && n[t(597)](typeof window.dispatchEvent, t(907)) && window[t(383)](new CustomEvent(n[t(898)], { detail: { launchSource: r } }));
  }
  [i(955)](e) {
    const t = i, n = { SzAtj: function(r, u) {
      return r != u;
    }, VZmPf: "undefined", yuNkk: function(r, u) {
      return r == u;
    }, EfAZG: t(907), FFddy: t(698), YOUOk: t(700) };
    let x = it[t(916)](e);
    n[t(622)](typeof window, n[t(436)]) && n[t(465)](typeof window[t(383)], n[t(1123)]) && window[t(383)](new CustomEvent(n[t(591)], { detail: x })), console.log(n[t(740)], x);
  }
  [i(532)](e) {
    const t = i, n = { pXkSE: function(r, u) {
      return r != u;
    }, AaXsE: function(r, u) {
      return r == u;
    }, uTpfk: t(907), nIeLD: t(759) };
    let x = Yt(e);
    n[t(900)](typeof window, "undefined") && n.AaXsE(typeof window.dispatchEvent, n[t(1099)]) && window[t(383)](new CustomEvent(n.nIeLD, { detail: x })), console[t(545)]("[EvenAppBridge] EvenHub event:", x);
  }
  [i(773)](e, t) {
    const n = i, x = { QKeEi: function(r, u) {
      return r !== u;
    }, HimBw: n(1079), IiXQZ: "call_even_app_method", SWlLW: function(r, u) {
      return r(u);
    } };
    return m(this, null, function* () {
      const r = n;
      return x.QKeEi(x[r(1008)], x[r(1008)]) ? _0x5ec353 ? _0x50072b instanceof _0x346f21 ? _0x4ca635[r(1128)]() : new _0x3b15f0(_0x405510)[r(1128)]() : {} : yield this.postMessage({ type: x[r(694)], method: x[r(1076)](String, e), data: t });
    });
  }
  getUserInfo() {
    const e = i, t = { QfCXN: e(1080), MrFQp: function(n, x, r, u) {
      return n(x, r, u);
    } };
    return t.MrFQp(m, this, null, function* () {
      const n = e;
      let x = yield this[n(773)](t[n(410)]);
      return Bt.fromJson(x);
    });
  }
  [i(364)]() {
    const e = i, t = { QZCjA: function(n, x) {
      return n !== x;
    }, kfmIw: e(975), Jcwvh: function(n, x, r, u) {
      return n(x, r, u);
    } };
    return t[e(627)](m, this, null, function* () {
      const n = e, x = { xUrrZ: function(r, u) {
        return r instanceof u;
      } };
      if (t[n(1107)](t[n(980)], n(464))) {
        let r = yield this[n(773)](n(458));
        return r ? Ht[n(916)](r) : null;
      } else return _0x230ac8 ? x[n(990)](_0xd8671d, _0x4ea394) ? _0x5e12ac[n(1128)]() : new _0x48d084(_0x385ea6)[n(1128)]() : {};
    });
  }
  [i(1196)](e, t) {
    return m(this, null, function* () {
      let n = yield this.callEvenApp("setLocalStorage", { key: e, value: t });
      return n ?? !1;
    });
  }
  [i(1034)](e) {
    const t = i, n = { FwXeo: t(1034), IZpJS: function(x, r) {
      return x != r;
    }, RfaIz: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n.RfaIz(m, this, null, function* () {
      const x = t;
      let r = yield this[x(773)](n[x(721)], { key: e });
      return n[x(786)](r, null) ? r : "";
    });
  }
  [i(1141)](e) {
    const t = i, n = { tjtME: t(1141) };
    return m(this, null, function* () {
      const x = t;
      let r = zt.toJson(e);
      r[x(1115)] = window.__EVEN_HUB_APP_ID__;
      let u = yield this.callEvenApp(n[x(484)], r);
      return tt[x(476)](u);
    });
  }
  [i(390)](e) {
    const t = { qPZew: "rebuildPageContainer", WhYCE: function(n, x) {
      return n != x;
    } };
    return m(this, null, function* () {
      const n = L;
      let x = yield this[n(773)](t.qPZew, mt[n(1128)](e));
      return t[n(738)](x, null) ? x : !1;
    });
  }
  [i(646)](e) {
    const t = i, n = { joXlq: function(x, r, u, s, c, l) {
      return x(r, u, s, c, l);
    }, rJTRG: t(571), YTJsE: t(461), NqWLT: t(1037), qnAQk: t(987), gXZJJ: function(x, r, u, s) {
      return x(r, u, s);
    }, WdPtM: function(x, r) {
      return x === r;
    }, CTcZz: t(859), MWkQZ: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n.MWkQZ(m, this, null, function* () {
      const x = t;
      if (n.WdPtM(n[x(549)], n[x(549)])) {
        let r = yield this[x(773)]("updateImageRawData", Tt[x(1128)](e));
        return et.normalize(r);
      } else return _0x388a85(_0x75b391) ? new _0x120183({ iMUReportEn: n[x(677)](_0x5be780, _0x45d864, n.rJTRG, n[x(840)], n[x(1088)], n[x(865)]), reportFrq: n.gXZJJ(_0xb1d035, _0x1735ae, x(369), "ReportFrq") }) : new _0xc8c113();
    });
  }
  [i(1198)](e) {
    const t = { HdDjE: function(n, x) {
      return n != x;
    } };
    return m(this, null, function* () {
      const n = L;
      let x = yield this[n(773)]("textContainerUpgrade", yt.toJson(e));
      return t[n(480)](x, null) ? x : !1;
    });
  }
  audioControl(e) {
    const t = i, n = { WEXzW: function(x, r) {
      return x === r;
    }, luUbZ: t(601), IMeQx: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n[t(866)](m, this, null, function* () {
      const x = t;
      return n[x(357)](yield this[x(773)](n.luUbZ, { isOpen: e }), !0);
    });
  }
  [i(501)](e) {
    const t = i, n = { yJgKh: function(x, r) {
      return x === r;
    }, oZNoe: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n[t(502)](m, this, arguments, function* (x, r = 100) {
      const u = t;
      return n[u(398)](yield this[u(773)](u(501), bt[u(1128)](new bt({ iMUReportEn: x ? 1 : 0, reportFrq: r }))), !0);
    });
  }
  shutDownPageContainer(e = 0) {
    const t = i, n = { xVuoX: function(x, r, u) {
      return x(r, u);
    }, RZMed: function(x, r) {
      return x != r;
    }, usLbj: function(x, r) {
      return x === r;
    }, mtaas: t(781), mOBUu: t(661), akELA: function(x, r, u, s) {
      return x(r, u, s);
    } };
    return n[t(418)](m, this, null, function* () {
      const x = t;
      if (n[x(959)](n.mtaas, n[x(528)])) {
        let r = new ht({ exitMode: e }), u = yield this[x(773)](n[x(629)], ht[x(1128)](r));
        return n.RZMed(u, null) ? u : !1;
      } else return n.xVuoX(_0x4abe15, {}, n.RZMed(this, null) ? this : {});
    });
  }
  [i(739)](e) {
    const t = i, n = { LBDfJ: function(r, u) {
      return r != u;
    }, vDVOH: function(r, u) {
      return r(u);
    }, JMKbt: t(848), TLwMP: function(r, u) {
      return r == u;
    }, eTrsD: t(1004) };
    if (n[t(849)](typeof window, n[t(625)])) return () => {
    };
    let x = (r) => {
      const u = t;
      var s;
      let c = r;
      n[u(674)](s = c[u(1203)], null) && s.launchSource && n[u(1102)](e, c.detail.launchSource);
    };
    return window[t(650)]("evenAppLaunchSource", x), () => {
      const r = t;
      window[r(953)](n[r(1108)], x);
    };
  }
  [i(883)](e) {
    const t = i, n = { FpFBH: function(r, u) {
      return r(u);
    }, UUMTb: function(r, u) {
      return r instanceof u;
    }, MHmos: function(r, u) {
      return r === u;
    }, vlNqB: t(440), ZkPhx: t(696), Skhqo: t(698), jDHoB: function(r, u) {
      return r == u;
    }, LfOyT: t(1004) };
    if (n.jDHoB(typeof window, n[t(551)])) return () => {
    };
    let x = (r) => {
      const u = t;
      n[u(798)](e, r[u(1203)]);
    };
    return window.addEventListener(n[t(917)], x), () => {
      const r = t;
      if (n.MHmos(n[r(531)], n[r(972)])) return _0x176f16 ? n.UUMTb(_0x1e643d, _0x1b5061) ? _0x121bff[r(1128)]() : new _0x2322c5(_0x29ca43)[r(1128)]() : {};
      window[r(953)](n.Skhqo, x);
    };
  }
  onEvenHubEvent(e) {
    const t = i, n = { BAjjh: function(r, u) {
      return r === u;
    }, jeyso: t(899), qxOtZ: "evenHubEvent", tBTYC: t(1004) };
    if (typeof window == n[t(509)]) return () => {
    };
    let x = (r) => {
      const u = t;
      n[u(996)](n[u(613)], "gRYMv") ? _0x22ec37[u(953)]("evenAppLaunchSource", _0x414a2f) : e(r.detail);
    };
    return window[t(650)](n[t(542)], x), () => {
      const r = t;
      window[r(953)](n[r(542)], x);
    };
  }
};
function Gt() {
  const o = i, e = { QjGuZ: function(t, n) {
    return t === n;
  }, xTZEQ: o(403), ZwDLu: function(t, n) {
    return t(n);
  }, GUaYB: function(t, n) {
    return t !== n;
  }, jxpQW: o(706), jieZU: o(887), OjrZa: function(t, n) {
    return t == n;
  }, yUUFm: o(1004), SRRbd: o(1158), hslum: "Esisv", GVFTp: o(348), tYdTY: o(683) };
  return new Promise((t) => {
    const n = o, x = { jRNhg: function(r, u) {
      return e[L(875)](r, u);
    } };
    if (e[n(1167)](e[n(366)], e[n(814)])) {
      if (e.OjrZa(typeof window, e[n(967)])) throw new Error(e[n(468)]);
      let r = window[n(998)];
      if (r && r.ready) {
        if (e[n(1050)](e[n(977)], e[n(1176)])) return;
        t(r);
        return;
      }
      let u = () => {
        let c = window[n(998)];
        c && (c._ready = !0, x.jRNhg(t, c));
      };
      window[n(650)](e[n(726)], u, { once: !0 }), r && setTimeout(() => {
        const s = n;
        if (e[s(1050)](s(653), e[s(358)])) {
          _0x610e2b(_0x3a871c);
          return;
        } else r && (r[s(760)] = !0, e[s(875)](t, r));
      }, 100);
    } else return _0x36ef14.isRing(this[n(816)]);
  });
}
typeof window < "u" && Jt[i(1063)]();
i(910);
i(1152);
var Ft = /* @__PURE__ */ ((o) => (o.LIST = "list", o.TEXT = "text", o))(Ft || {});
class Zt {
  constructor(e, t) {
    return this.positionX = e, this.positionY = t, this;
  }
  setX(e) {
    return this.positionX = e, g.logger.debug(`[Element] Element position X set to ${e}.`), this;
  }
  setY(e) {
    return this.positionY = e, g.logger.debug(`[Element] Element position Y set to ${e}.`), this;
  }
  get x() {
    return this.positionX;
  }
  get y() {
    return this.positionY;
  }
}
class M {
  constructor(e, t) {
    return this.sizeWidth = e, this.sizeHeight = t, this;
  }
  static MAX_WIDTH = 576;
  static MAX_HEIGHT = 238;
  setWidth(e) {
    return e > M.MAX_WIDTH && (e = M.MAX_WIDTH, g.logger.warn(
      `[Element] Width is too large, clamped to ${M.MAX_WIDTH}.`
    )), this.sizeWidth = e, g.logger.debug(`[Element] Element width set to ${e}.`), this;
  }
  setHeight(e) {
    return e > M.MAX_HEIGHT && (e = M.MAX_HEIGHT, g.logger.warn(
      `[Element] Height is too large, clamped to ${M.MAX_HEIGHT}.`
    )), this.sizeHeight = e, g.logger.debug(`[Element] Element height set to ${e}.`), this;
  }
  get width() {
    return this.sizeWidth;
  }
  get height() {
    return this.sizeHeight;
  }
}
class Ct {
  constructor(e, t, n) {
    return this.borderWidth = e, this.borderColor = t, this.borderRadius = n, this;
  }
  setWidth(e) {
    return this.borderWidth = e, g.logger.debug(`[Element] Border width set to ${e}.`), this;
  }
  setColor(e) {
    return this.borderColor = e, g.logger.debug(`[Element] Border color set to ${e}.`), this;
  }
  setRadius(e) {
    return this.borderRadius = e, g.logger.debug(`[Element] Border radius set to ${e}.`), this;
  }
  get width() {
    return this.borderWidth;
  }
  get color() {
    return this.borderColor;
  }
  get radius() {
    return this.borderRadius;
  }
}
class xt {
  constructor(e, t) {
    return this.page = e, this.type = t, g.logger.debug(
      `[Element] Element "${this.containerID}" created with type "${this.type}".`
    ), this;
  }
  static elementIdCounter = 0;
  containerID = xt.elementIdCounter++;
  position = new Zt(0, 0);
  size = new M(100, 100);
  afterRender() {
    return Promise.resolve();
  }
  get id() {
    return this.containerID;
  }
  toEvenSdkElement() {
    return g.logger.debug(`[Element] Serializing element "${this.containerID}".`), {
      containerID: this.containerID,
      containerName: this.id.toString(),
      xPosition: this.position.x,
      yPosition: this.position.y,
      width: this.size.width,
      height: this.size.height
    };
  }
  markAsEventCaptureElement() {
    return g.logger.info(`[Element] Element "${this.containerID}" marked as event capture.`), this.page.setEventCaptureElement(this), this;
  }
  setPosition(e) {
    return e(this.position), g.logger.debug(`[Element] Element "${this.containerID}" position updated.`), this;
  }
  setSize(e) {
    return e(this.size), g.logger.debug(`[Element] Element "${this.containerID}" size updated.`), this;
  }
}
class ft extends xt {
}
class Vt extends xt {
  border = null;
  items = [];
  itemWidth = 100;
  isItemSelectBorderEn = !1;
  constructor(e, t) {
    return super(
      e,
      "list"
      /* LIST */
    ), this.items = t, this;
  }
  isDirty = !0;
  async afterRender() {
    this.isDirty = !1, g.logger.debug(`[Element] List element "${this.id}" rendered.`);
  }
  get didChange() {
    return this.isDirty;
  }
  addItem(e) {
    return this.items.push(e), this.isDirty = !0, g.logger.debug(`[Element] List element "${this.id}" added item.`), this;
  }
  removeItem(e) {
    return this.items.splice(e, 1), this.isDirty = !0, g.logger.debug(`[Element] List element "${this.id}" removed item at ${e}.`), this;
  }
  setItems(e) {
    return this.items = e, this.isDirty = !0, g.logger.debug(
      `[Element] List element "${this.id}" items set (${e.length} total).`
    ), this;
  }
  replaceItem(e, t) {
    return this.items[e] = t, this.isDirty = !0, g.logger.debug(`[Element] List element "${this.id}" replaced item at ${e}.`), this;
  }
  toEvenSdkElement() {
    return g.logger.debug(`[Element] Serializing list element "${this.id}".`), O.fromJson({
      ...super.toEvenSdkElement(),
      borderWidth: this.border?.width,
      borderColor: this.border?.color,
      borderRdaius: this.border?.radius,
      itemContainer: ct.fromJson({
        itemCount: this.items.length,
        itemWidth: this.itemWidth,
        isItemSelectBorderEn: this.isItemSelectBorderEn ? 1 : 0,
        itemName: this.items
      })
    });
  }
  setBorder(e) {
    return this.border === null && (this.border = new Ct(0, "#000000", 0)), e(this.border), this.isDirty = !0, g.logger.debug(`[Element] List element "${this.id}" border updated.`), this;
  }
  setItemWidth(e) {
    return this.itemWidth = e, this.isDirty = !0, g.logger.debug(`[Element] List element "${this.id}" item width set to ${e}.`), this;
  }
  setIsItemSelectBorderEn(e) {
    return this.isItemSelectBorderEn = e, this.isDirty = !0, g.logger.debug(
      `[Element] List element "${this.id}" item select border set to ${e}.`
    ), this;
  }
}
class Qt extends ft {
  constructor(e, t) {
    return super(
      e,
      "text"
      /* TEXT */
    ), this.content = t, this;
  }
  border = null;
  isDirty = !0;
  async afterRender() {
    this.isDirty = !1, g.logger.debug(`[Element] Text element "${this.id}" rendered.`);
  }
  toEvenSdkElement() {
    return g.logger.debug(`[Element] Serializing text element "${this.id}".`), X.fromJson({
      ...super.toEvenSdkElement(),
      borderWidth: this.border?.width,
      borderColor: this.border?.color,
      borderRdaius: this.border?.radius,
      content: this.content
    });
  }
  get didChange() {
    return this.isDirty;
  }
  setBorder(e) {
    return this.border === null && (this.border = new Ct(0, "#000000", 0)), e(this.border), this.isDirty = !0, g.logger.debug(`[Element] Text element "${this.id}" border updated.`), this;
  }
  setContent(e) {
    return this.content = e, this.isDirty = !0, g.logger.debug(`[Element] Text element "${this.id}" content updated.`), this;
  }
  async updateWithEvenHubSdk() {
    g.logger.info(`[Element] Updating text element "${this.id}" via Even hub SDK.`);
    try {
      const t = await (await g.getRawBridge()).textContainerUpgrade(yt.fromJson({
        containerID: this.id,
        containerName: this.id.toString(),
        content: this.content
      }));
      return this.isDirty = !1, g.logger.debug(`[Element] Text element "${this.id}" update result: ${t}.`), t;
    } catch (e) {
      throw g.logger.error(`[Element] Text element "${this.id}" update failed.`), e;
    }
  }
}
let $t = (o = 21) => crypto.getRandomValues(new Uint8Array(o)).reduce((e, t) => (t &= 63, t < 36 ? e += t.toString(36) : t < 62 ? e += (t - 26).toString(36).toUpperCase() : t > 62 ? e += "-" : e += "_", e), "");
class kt {
  constructor(e) {
    return this.sdk = e, this.id = $t(), g.logger.info(`[Page] EvenBetterPage created with id "${this.id}".`), this;
  }
  id;
  elements = /* @__PURE__ */ new Map();
  eventCaptureElementId = null;
  async render() {
    return g.logger.info(`[Page] Rendering EvenBetterPage "${this.id}".`), this.sdk.renderPage(this);
  }
  toEvenSdkPage() {
    const e = Array.from(this.elements.values());
    return g.logger.debug(
      `[Page] Building SDK page for "${this.id}" with ${e.length} elements.`
    ), zt.fromJson({
      containerTotalNum: this.elements.size,
      listObject: e.filter((t) => t.type === "list").map((t) => ({
        ...t.toEvenSdkElement(),
        isEventCapture: t.id === this.eventCaptureElementId ? 1 : 0
      })),
      textObject: e.filter((t) => t.type === "text").map((t) => ({
        ...t.toEvenSdkElement(),
        isEventCapture: t.id === this.eventCaptureElementId ? 1 : 0
      }))
    });
  }
  getElements() {
    return g.logger.debug(`[Page] Getting elements for page "${this.id}".`), Array.from(this.elements.values());
  }
  setEventCaptureElement(e) {
    g.logger.info(
      `[Page] Setting event capture element "${e.id}" for page "${this.id}".`
    ), this.eventCaptureElementId = e.id;
  }
  addTextElement(e) {
    g.logger.info(`[Page] Adding text element to page "${this.id}".`);
    const t = new Qt(this, e);
    return this.elements.set(t.id, t), g.logger.debug(
      `[Page] Text element "${t.id}" added to page "${this.id}".`
    ), t;
  }
  addListElement(e) {
    g.logger.info(`[Page] Adding list element to page "${this.id}".`);
    const t = new Vt(this, e);
    return this.elements.set(t.id, t), g.logger.debug(
      `[Page] List element "${t.id}" added to page "${this.id}".`
    ), t;
  }
}
class y {
  static level = "error";
  static implementation = {
    warn: console.warn,
    error: console.error,
    info: console.info,
    debug: console.debug
  };
  static debug(e) {
    y.level === "debug" && y.implementation.debug(e);
  }
  static info(e) {
    ["debug", "info"].includes(y.level) && y.implementation.info(e);
  }
  static warn(e) {
    ["debug", "info", "warn"].includes(y.level) && y.implementation.warn(e);
  }
  static error(e) {
    y.level !== "none" && y.implementation.error(e);
  }
}
const te = () => {
  const o = globalThis;
  return o.__evenBetterSdkSharedState || (o.__evenBetterSdkSharedState = {
    bridge: null,
    initializationPromise: null,
    currentPageId: null,
    eventListeners: [],
    renderRequests: [],
    renderInProgress: !1
  }), o.__evenBetterSdkSharedState;
};
class g {
  pages = /* @__PURE__ */ new Map();
  static logger = y;
  static get state() {
    return te();
  }
  static get bridge() {
    return g.state.bridge;
  }
  static set bridge(e) {
    g.state.bridge = e;
  }
  static get currentPageId() {
    return g.state.currentPageId;
  }
  static set currentPageId(e) {
    g.state.currentPageId = e;
  }
  static get eventListeners() {
    return g.state.eventListeners;
  }
  static get renderRequests() {
    return g.state.renderRequests;
  }
  static get renderInProgress() {
    return g.state.renderInProgress;
  }
  static set renderInProgress(e) {
    g.state.renderInProgress = e;
  }
  static get initializationPromise() {
    return g.state.initializationPromise;
  }
  static set initializationPromise(e) {
    g.state.initializationPromise = e;
  }
  static setLogger(e) {
    y.implementation = e;
  }
  static setLogLevel(e) {
    y.level = e;
  }
  createPage(e) {
    g.logger.info(`[SDK] Creating page for identifier "${e}".`);
    const t = new kt(this);
    return this.pages.set(e, t), g.logger.debug(`[SDK] Page created with id "${t.id}" for identifier "${e}".`), t;
  }
  get isInitialized() {
    return g.logger.debug(`[SDK] SDK initialized: ${g.bridge !== null}.`), g.bridge !== null;
  }
  static async initialize() {
    if (g.initializationPromise && (g.logger.debug("[SDK] SDK already initializing, waiting for completion."), await g.initializationPromise), g.bridge) {
      g.logger.debug("[SDK] SDK already initialized, skipping setup.");
      return;
    }
    let e = () => {
      throw new Error("Initialization promise called before it was set");
    };
    g.initializationPromise = new Promise((t) => {
      e = t;
    });
    try {
      g.logger.info("[SDK] Initializing EvenBetterSdk bridge.");
      const t = await Gt();
      g.logger.debug("[SDK] Even app bridge detected, waiting for readiness.");
      let n = 0;
      for (; !t.ready; )
        await new Promise((x) => setTimeout(x, 100)), n += 1, n % 10 === 0 && g.logger.info(`[SDK] Still waiting for bridge.ready (${n * 100}ms elapsed).`);
      g.logger.info("[SDK] Even app bridge is ready."), t.onEvenHubEvent((x) => {
        g.logger.debug("[SDK] Even hub event received, dispatching to listeners."), this.eventListeners.forEach((r) => r(x));
      }), this.currentPageId = null, this.bridge = t, g.logger.info("[SDK] EvenBetterSdk bridge initialization complete.");
    } finally {
      e();
    }
  }
  async getValue(e) {
    return await g.initialize(), g.bridge.getLocalStorage(e);
  }
  async setValue(e, t) {
    await g.initialize(), await g.bridge.setLocalStorage(e, t);
  }
  static async getRawBridge() {
    return await g.initialize(), g.bridge;
  }
  async renderPage(e) {
    g.logger.debug(`[SDK] adding render request for page "${e.id}".`);
    const t = new Promise((n) => {
      g.renderRequests.push({ page: e, resolve: n });
    });
    g.logger.debug("[SDK] starting to process render requests."), g.processRenderRequests(), g.logger.debug("[SDK] waiting for render request to complete."), await t, g.logger.debug(`[SDK] render request for page "${e.id}" completed.`);
  }
  static async processRenderRequests() {
    if (g.renderInProgress)
      return;
    g.renderInProgress = !0;
    const e = g.renderRequests.shift();
    if (!e) {
      g.renderInProgress = !1;
      return;
    }
    this.logger.debug(`[SDK] Processing render request for page "${e.page.id}".`);
    const { page: t, resolve: n } = e;
    try {
      g.logger.debug(`[SDK] Rendering page "${t.id}".`), await g.initialize();
      const x = t.getElements();
      g.logger.debug(`[SDK] Page "${t.id}" has ${x.length} elements.`);
      const r = x.filter((d) => d.didChange ? d instanceof ft : !1), s = x.filter((d) => d.didChange).some(
        (d) => !(d instanceof ft)
      ), c = g.currentPageId === t.id, l = !c || s;
      if (g.logger.debug(
        `[SDK] Page "${t.id}" requires full render: ${l} (Same page: ${c}).`
      ), l) {
        g.currentPageId === null ? (g.logger.info(`[SDK] Creating startup page container for "${t.id}".`), await g.bridge.createStartUpPageContainer(t.toEvenSdkPage())) : (g.logger.info(`[SDK] Rebuilding page container for "${t.id}".`), await g.bridge.rebuildPageContainer(mt.fromJson(t.toEvenSdkPage()))), g.currentPageId = t.id, g.logger.debug(`[SDK] Page "${t.id}" render complete with full rebuild.`);
        return;
      }
      for (const d of r) {
        g.logger.debug(`[SDK] Updating element "${d.id}" with partial update.`);
        try {
          const a = await d.updateWithEvenHubSdk();
          g.logger.debug(`[SDK] Element "${d.id}" update result: ${a}.`);
        } catch (a) {
          throw g.logger.error(`[SDK] Element "${d.id}" partial update failed.`), a;
        }
      }
      g.logger.debug(`[SDK] Page "${t.id}" render complete with partial updates.`);
    } finally {
      g.logger.debug(`[SDK] Resolving render request for page "${t.id}".`), n(), g.renderInProgress = !1, setTimeout(() => {
        g.processRenderRequests();
      }, 0);
    }
  }
  addEventListener(e) {
    g.logger.debug("[SDK] Adding Even hub event listener."), g.eventListeners.push(e);
  }
  removeEventListener(e) {
    g.logger.debug("[SDK] Removing Even hub event listener."), g.eventListeners.splice(g.eventListeners.indexOf(e), 1);
  }
}
export {
  xt as EvenBetterElement,
  Zt as EvenBetterElementPosition,
  M as EvenBetterElementSize,
  Ft as EvenBetterElementType,
  ft as EvenBetterElementWithPartialUpdate,
  Vt as EvenBetterListElement,
  kt as EvenBetterPage,
  g as EvenBetterSdk,
  Qt as EvenBetterTextElement
};
